diff -ur lame-3.100/ACM/main.cpp lame-3.100+iPhone/ACM/main.cpp
--- lame-3.100/ACM/main.cpp	2006-12-25 11:37:34.000000000 -1000
+++ lame-3.100+iPhone/ACM/main.cpp	2018-07-27 22:55:45.000000000 -1000
@@ -156,7 +156,7 @@
 			if (debug != NULL)
 			{
 //				debug->OutPut(DEBUG_LEVEL_MSG, "DRV_LOAD, version %s %s %s", ACM_VERSION, __DATE__, __TIME__);
-				debug->OutPut(DEBUG_LEVEL_MSG, "DRV_LOAD, %s %s",  __DATE__, __TIME__);
+				debug->OutPut(DEBUG_LEVEL_MSG, "DRV_LOAD");
 			}
 			return 1L;
 		}
diff -ur lame-3.100/frontend/gtkanal.c lame-3.100+iPhone/frontend/gtkanal.c
--- lame-3.100/frontend/gtkanal.c	2017-09-06 09:33:35.000000000 -1000
+++ lame-3.100+iPhone/frontend/gtkanal.c	2018-07-27 22:56:05.000000000 -1000
@@ -1128,11 +1128,11 @@
 #if   MP3X_ALPHA_VERSION > 0
     static /*@observer@ */ const char *const str =
         XSTR(MP3X_MAJOR_VERSION) "." XSTR(MP3X_MINOR_VERSION)
-        " (alpha " XSTR(MP3X_ALPHA_VERSION) ", " __DATE__ " " __TIME__ ")";
+        " (alpha " XSTR(MP3X_ALPHA_VERSION) ")";
 #elif MP3X_BETA_VERSION > 0
     static /*@observer@ */ const char *const str =
         XSTR(MP3X_MAJOR_VERSION) "." XSTR(MP3X_MINOR_VERSION)
-        " (beta " XSTR(MP3X_BETA_VERSION) ", " __DATE__ ")";
+        " (beta " XSTR(MP3X_BETA_VERSION) ")";
 #else
     static /*@observer@ */ const char *const str =
         XSTR(MP3X_MAJOR_VERSION) "." XSTR(MP3X_MINOR_VERSION);
diff -ur lame-3.100/libmp3lame/version.c lame-3.100+iPhone/libmp3lame/version.c
--- lame-3.100/libmp3lame/version.c	2011-11-17 23:51:02.000000000 -1000
+++ lame-3.100+iPhone/libmp3lame/version.c	2018-07-27 22:55:08.000000000 -1000
@@ -58,11 +58,11 @@
 #if   LAME_ALPHA_VERSION
     static /*@observer@ */ const char *const str =
         STR(LAME_MAJOR_VERSION) "." STR(LAME_MINOR_VERSION) " "
-        "(alpha " STR(LAME_PATCH_VERSION) ", " __DATE__ " " __TIME__ ")";
+        "(alpha " STR(LAME_PATCH_VERSION)")";
 #elif LAME_BETA_VERSION
     static /*@observer@ */ const char *const str =
         STR(LAME_MAJOR_VERSION) "." STR(LAME_MINOR_VERSION) " "
-        "(beta " STR(LAME_PATCH_VERSION) ", " __DATE__ ")";
+        "(beta " STR(LAME_PATCH_VERSION) ")";
 #elif LAME_RELEASE_VERSION && (LAME_PATCH_VERSION > 0)
     static /*@observer@ */ const char *const str =
         STR(LAME_MAJOR_VERSION) "." STR(LAME_MINOR_VERSION) "." STR(LAME_PATCH_VERSION);
@@ -165,11 +165,11 @@
 #if   PSY_ALPHA_VERSION > 0
     static /*@observer@ */ const char *const str =
         STR(PSY_MAJOR_VERSION) "." STR(PSY_MINOR_VERSION)
-        " (alpha " STR(PSY_ALPHA_VERSION) ", " __DATE__ " " __TIME__ ")";
+        " (alpha " STR(PSY_ALPHA_VERSION) ")";
 #elif PSY_BETA_VERSION > 0
     static /*@observer@ */ const char *const str =
         STR(PSY_MAJOR_VERSION) "." STR(PSY_MINOR_VERSION)
-        " (beta " STR(PSY_BETA_VERSION) ", " __DATE__ ")";
+        " (beta " STR(PSY_BETA_VERSION) ")";
 #else
     static /*@observer@ */ const char *const str =
         STR(PSY_MAJOR_VERSION) "." STR(PSY_MINOR_VERSION);
