diff -ru sniffit.0.3.7.beta/configure.in sniffit.0.3.7.beta+iPhone/configure.in
--- sniffit.0.3.7.beta/configure.in	1998-07-16 16:17:10.000000000 +0000
+++ sniffit.0.3.7.beta+iPhone/configure.in	2009-02-26 13:38:35.000000000 +0000
@@ -6,7 +6,7 @@
 AC_INIT(sniffit.h)
 
 dnl Config of libpcap
-AC_CONFIG_SUBDIRS(libpcap)
+#AC_CONFIG_SUBDIRS(libpcap)
 
 dnl Check the C compiler
 AC_PROG_CC
@@ -100,6 +100,11 @@
 	OS_OPT=
 	AC_SUBST(OS_OPT)
         ;;
+darwin*)
+        AC_DEFINE(DARWIN)
+	OS_OPT=
+	AC_SUBST(OS_OPT)
+        ;;
 *)
 	echo "NOT A SUPPORTED SYSTEM / OR SYSTEM NOT RECOGNISED"
 	echo "Contact <Coder@reptile.rug.ac.be> if you feel it might be a bug."
diff -ru sniffit.0.3.7.beta/sn_data.h sniffit.0.3.7.beta+iPhone/sn_data.h
--- sniffit.0.3.7.beta/sn_data.h	1998-07-16 16:17:10.000000000 +0000
+++ sniffit.0.3.7.beta+iPhone/sn_data.h	2009-02-26 13:37:29.000000000 +0000
@@ -60,6 +60,12 @@
 int HEADSIZE[]={14}; 
 #endif
 
+#ifdef DARWIN
+#define NETDEV_NR      1
+char *NETDEV[]={"en", "lo"};
+int HEADSIZE[]={14, 4}; 
+#endif
+
 #ifdef NETBSD
 #ifdef i386
 #define NETDEV_NR      21
