diff -ru p7zip_4.57/CPP/Windows/DLL.cpp p7zip_4.57+iPhone/CPP/Windows/DLL.cpp
--- p7zip_4.57/CPP/Windows/DLL.cpp	2007-11-14 21:07:44.000000000 +0000
+++ p7zip_4.57+iPhone/CPP/Windows/DLL.cpp	2008-06-18 02:49:22.000000000 +0000
@@ -2,7 +2,7 @@
 
 #include "StdAfx.h"
 
-#ifdef __APPLE_CC__
+#if 0//__APPLE_CC__
 #include <mach-o/dyld.h>
 #elif ENV_BEOS
 #include <kernel/image.h>
@@ -39,7 +39,7 @@
   if (_module == 0)
     return true;
 
-#ifdef __APPLE_CC__
+#if 0//__APPLE_CC__
   int ret = NSUnLinkModule ((NSModule)_module, 0);
 #elif ENV_BEOS
   int ret = unload_add_on((image_id)_module);
@@ -57,7 +57,7 @@
   void *ptr = 0;
   TRACEN((printf("local_GetProcAddress(%p,%s)\n",(void *)module,lpProcName)))
   if (module) {
-#ifdef __APPLE_CC__
+#if 0//__APPLE_CC__
     char name[MAX_PATHNAME_LEN];
     snprintf(name,sizeof(name),"_%s",lpProcName);
     name[sizeof(name)-1] = 0;
@@ -117,7 +117,7 @@
     strcpy(name+len-4,".so");
   }
 
-#ifdef __APPLE_CC__
+#if 0//__APPLE_CC__
   NSObjectFileImage image;
   NSObjectFileImageReturnCode nsret;
 
@@ -171,7 +171,7 @@
     if (fctTest) fctTest();
 
   } else {
-#ifdef __APPLE_CC__
+#if 0//__APPLE_CC__
     NSLinkEditErrors c;
     int num_err;
     const char *file,*err;
