diff -ru classpath-0.99/native/jni/java-io/java_io_VMConsole.c classpath-0.99+iPhone/native/jni/java-io/java_io_VMConsole.c
--- classpath-0.99/native/jni/java-io/java_io_VMConsole.c	2012-01-08 12:35:53.000000000 +0000
+++ classpath-0.99+iPhone/native/jni/java-io/java_io_VMConsole.c	2016-01-06 10:13:45.000000000 +0000
@@ -47,6 +47,10 @@
 
 /*************************************************************************/
 
+#ifndef IUCLC
+#define IUCLC 0
+#endif
+
 #define TERMIOS_ECHO_IFLAGS (IUCLC|IXON|IXOFF|IXANY)
 #define TERMIOS_ECHO_LFLAGS (ECHO|ECHOE|ECHOK|ECHONL|TOSTOP)
 
