diff -ur dpkg-1.18.25/configure dpkg-1.18.25+iPhone/configure
--- dpkg-1.18.25/configure	2018-06-26 00:28:08.000000000 -1000
+++ dpkg-1.18.25+iPhone/configure	2018-09-14 10:14:00.000000000 -1000
@@ -909,6 +909,7 @@
 CPP
 LT_SYS_LIBRARY_PATH
 PERL
+LIBDIR_PERL
 PERL_LIBDIR
 TAR
 CXX
@@ -15450,7 +15451,7 @@
     # Let the user override the variable.
   if test -z "$PERL_LIBDIR"; then :
 
-    PERL_LIBDIR=$($PERL -MConfig -e \
+    PERL_LIBDIR=$($LIBDIR_PERL -MConfig -e \
                         'my $r = $Config{vendorlibexp};
                          $r =~ s/$Config{vendorprefixexp}/\$(prefix)/;
                          print $r')
