diff -ur dpkg-1.18.25/lib/dpkg/tarfn.c dpkg-1.18.25+iPhone/lib/dpkg/tarfn.c
--- dpkg-1.18.25/lib/dpkg/tarfn.c	2018-06-25 16:48:17.000000000 -1000
+++ dpkg-1.18.25+iPhone/lib/dpkg/tarfn.c	2018-11-14 21:23:25.000000000 -1000
@@ -345,7 +345,8 @@
 	int long_read;
 
 	free(*longp);
-	*longp = bp = m_malloc(te->size);
+	*longp = bp = m_malloc(te->size+1);
+	bzero(bp, te->size+1);
 
 	for (long_read = te->size; long_read > 0; long_read -= TARBLKSZ) {
 		int copysize;
