diff -ur dpkg-1.18.25/data/tupletable dpkg-1.18.25+iPhone/data/tupletable
--- dpkg-1.18.25/data/tupletable	2018-06-17 05:49:22.000000000 -1000
+++ dpkg-1.18.25+iPhone/data/tupletable	2018-09-14 10:22:50.000000000 -1000
@@ -33,6 +33,8 @@
 base-bsd-openbsd-<cpu>		openbsd-<cpu>
 base-bsd-netbsd-<cpu>		netbsd-<cpu>
 base-bsd-darwin-<cpu>		darwin-<cpu>
+base-bsd-darwin-arm		iphoneos-arm
+base-bsd-darwin-arm64		iphoneos-arm
 base-sysv-aix-<cpu>		aix-<cpu>
 base-sysv-solaris-<cpu>		solaris-<cpu>
 eabi-uclibc-uclinux-arm		uclinux-armel
