diff -ur libvncserver-LibVNCServer-0.9.11/Makefile.am libvncserver-LibVNCServer-0.9.11+iPhone/Makefile.am
--- libvncserver-LibVNCServer-0.9.11/Makefile.am	2016-12-30 03:01:28.000000000 -1000
+++ libvncserver-LibVNCServer-0.9.11+iPhone/Makefile.am	2018-08-29 11:17:20.000000000 -1000
@@ -1,7 +1,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS=libvncserver examples libvncclient webclients client_examples test
-DIST_SUBDIRS=libvncserver examples libvncclient webclients client_examples test
+SUBDIRS=libvncserver libvncclient test
+DIST_SUBDIRS=libvncserver libvncclient test
 EXTRA_DIST = CMakeLists.txt rfb/rfbconfig.h.cmake
 
 bin_SCRIPTS = libvncserver-config
