diff -ru pyobjc-1946/pyobjc-core/Modules/objc/pointer-support.m pyobjc-1946+iPhone/pyobjc-core/Modules/objc/pointer-support.m
--- pyobjc-1946/pyobjc-core/Modules/objc/pointer-support.m	2007-11-01 13:04:34.000000000 +0000
+++ pyobjc-1946+iPhone/pyobjc-core/Modules/objc/pointer-support.m	2008-03-27 10:59:08.000000000 +0000
@@ -22,6 +22,7 @@
 
 
 #import <CoreFoundation/CoreFoundation.h>
+#import <CFNetwork/CFNetwork.h>
 
 struct wrapper {
 	const char* signature;
