diff -ur curl-7.61.0/acinclude.m4 curl-7.61.0+iPhone/acinclude.m4
--- curl-7.61.0/acinclude.m4	2018-07-08 20:42:12.000000000 -1000
+++ curl-7.61.0+iPhone/acinclude.m4	2018-07-30 23:44:08.000000000 -1000
@@ -2615,18 +2615,6 @@
   AC_MSG_RESULT([$tst_cflags]);
 
   if test "$tst_cflags" = "yes"; then
-    AC_MSG_CHECKING([for *version-min in CFLAGS])
-    min=""
-    if test -z "$(echo $CFLAGS | grep m.*os.*-version-min)"; then
-      min="-mmacosx-version-min=10.8"
-      CFLAGS="$CFLAGS $min"
-    fi
-    if test -z "$min"; then
-      AC_MSG_RESULT([set by user])
-    else
-      AC_MSG_RESULT([$min set])
-    fi
-
     old_CFLAGS=$CFLAGS
     CFLAGS="$CFLAGS -Werror=partial-availability"
     AC_MSG_CHECKING([whether $CC accepts -Werror=partial-availability])
diff -ur curl-7.61.0/configure curl-7.61.0+iPhone/configure
--- curl-7.61.0/configure	2018-07-08 20:52:18.000000000 -1000
+++ curl-7.61.0+iPhone/configure	2018-07-30 23:44:40.000000000 -1000
@@ -19046,21 +19046,6 @@
 $as_echo "$tst_cflags" >&6; };
 
   if test "$tst_cflags" = "yes"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for *version-min in CFLAGS" >&5
-$as_echo_n "checking for *version-min in CFLAGS... " >&6; }
-    min=""
-    if test -z "$(echo $CFLAGS | grep m.*os.*-version-min)"; then
-      min="-mmacosx-version-min=10.8"
-      CFLAGS="$CFLAGS $min"
-    fi
-    if test -z "$min"; then
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: set by user" >&5
-$as_echo "set by user" >&6; }
-    else
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: $min set" >&5
-$as_echo "$min set" >&6; }
-    fi
-
     old_CFLAGS=$CFLAGS
     CFLAGS="$CFLAGS -Werror=partial-availability"
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC accepts -Werror=partial-availability" >&5
