--- mesa-19.3.1/meson.build	2019-12-18 10:36:00.000000000 -0800
+++ mesa-19.3.1+iPhone/meson.build	2020-01-10 09:16:58.000000000 -0800
@@ -256,19 +256,7 @@
   error('tegra driver requires nouveau driver')
 endif
 
-if host_machine.system() == 'darwin'
-  with_dri_platform = 'apple'
-  pre_args += '-DBUILDING_MESA'
-elif ['windows', 'cygwin'].contains(host_machine.system())
-  with_dri_platform = 'windows'
-elif system_has_kms_drm
-  with_dri_platform = 'drm'
-else
-  # FIXME: haiku doesn't use dri, and xlib doesn't use dri, probably should
-  # assert here that one of those cases has been met.
-  # FIXME: illumos ends up here as well
-  with_dri_platform = 'none'
-endif
+with_dri_platform = 'none'
 
 _platforms = get_option('platforms')
 if _platforms.contains('auto')
--- mesa-19.3.1/include/c11/threads_posix.h	2019-12-18 10:36:00.000000000 -0800
+++ mesa-19.3.1+iPhone/include/c11/threads_posix.h	2020-01-10 09:38:47.000000000 -0800
@@ -382,7 +382,7 @@
 
 /*-------------------- 7.25.7 Time functions --------------------*/
 // 7.25.6.1
-#ifndef HAVE_TIMESPEC_GET
+#if ! defined(HAVE_TIMESPEC_GET) || defined(__APPLE__) 
 static inline int
 timespec_get(struct timespec *ts, int base)
 {
--- mesa-19.3.1/src/mesa/main/texcompress_s3tc_tmp.h	2019-12-18 10:36:01.000000000 -0800
+++ mesa-19.3.1+iPhone/src/mesa/main/texcompress_s3tc_tmp.h	2020-01-10 09:47:45.000000000 -0800
@@ -25,11 +25,7 @@
 #ifndef TEXCOMPRESS_S3TC_TMP_H
 #define TEXCOMPRESS_S3TC_TMP_H
 
-#ifdef __APPLE__
-#include <OpenGL/gl.h>
-#else
 #include <GL/gl.h>
-#endif
 
 typedef GLubyte GLchan;
 #define UBYTE_TO_CHAN(b)  (b)
--- mesa-19.3.1/scons/x11.py	2019-12-18 10:36:00.000000000 -0800
+++ mesa-19.3.1+iPhone/scons/x11.py	2020-01-10 10:27:39.000000000 -0800
@@ -29,8 +29,6 @@
 
 
 def generate(env):
-    env.Append(CPPPATH = ['/usr/X11R6/include'])
-    env.Append(LIBPATH = ['/usr/X11R6/lib'])
 
     env.Append(LIBS = [
         'X11',
