diff -ur perl-5.28.0/dist/IO/poll.h perl-5.28.0+iPhone/dist/IO/poll.h
--- perl-5.28.0/dist/IO/poll.h	2018-05-20 22:41:35.000000000 -1000
+++ perl-5.28.0+iPhone/dist/IO/poll.h	2018-08-16 22:30:35.000000000 -1000
@@ -10,6 +10,8 @@
 #ifndef POLL_H
 #  define POLL_H
 
+#include <sys/poll.h>
+#if 0
 #if (defined(HAS_POLL) && defined(I_POLL)) || defined(POLLWRBAND)
 #  include <poll.h>
 #elif (defined(HAS_POLL) && defined(I_SYS_POLL))
@@ -58,5 +60,5 @@
 
 #endif /* I_POLL */
 
+#endif /* 0 */
 #endif /* POLL_H */
-
