diff -ur afpfs-ng-0.8.1/cmdline/cmdline_afp.h afpfs-ng-0.8.1+iPhone/cmdline/cmdline_afp.h
--- afpfs-ng-0.8.1/cmdline/cmdline_afp.h	2008-02-17 17:51:20.000000000 -1000
+++ afpfs-ng-0.8.1+iPhone/cmdline/cmdline_afp.h	2018-08-01 15:49:36.000000000 -1000
@@ -1,7 +1,7 @@
 #ifndef CMDLINE_AFP_H_
 #define CMDLINE_AFP_H_
 
-int full_url;
+extern int full_url;
 
 int com_connect(char * arg);
 int com_dir(char * arg);
