diff -ur wget-1.19.5/lib/localtime-buffer.c wget-1.19.5+iPhone/lib/localtime-buffer.c
--- wget-1.19.5/lib/localtime-buffer.c	2018-01-19 11:19:16.000000000 -1000
+++ wget-1.19.5+iPhone/lib/localtime-buffer.c	2018-09-15 12:53:52.000000000 -1000
@@ -23,6 +23,8 @@
 #include "localtime-buffer.h"
 
 #if GETTIMEOFDAY_CLOBBERS_LOCALTIME || TZSET_CLOBBERS_LOCALTIME
+#undef localtime
+#undef gmtime
 
 static struct tm tm_zero_buffer;
 struct tm *localtime_buffer_addr = &tm_zero_buffer;
