diff -ru faac/libfaac/frame.c faac+iPhone/libfaac/frame.c
--- faac/libfaac/frame.c	2004-11-17 14:26:06.000000000 +0000
+++ faac+iPhone/libfaac/frame.c	2008-06-18 00:50:35.000000000 +0000
@@ -51,7 +51,7 @@
 #if FAAC_RELEASE
 static char *libfaacName = FAAC_VERSION;
 #else
-static char *libfaacName = FAAC_VERSION ".1 (" __DATE__ ") UNSTABLE";
+static char *libfaacName = FAAC_VERSION ".1 UNSTABLE";
 #endif
 static char *libCopyright =
   "FAAC - Freeware Advanced Audio Coder (http://www.audiocoding.com/)\n"
