--- libvpx-1.8.0/build/make/Makefile	2019-02-04 17:02:33.000000000 +0000
+++ libvpx-1.8.0+iPhone/build/make/Makefile	2019-04-04 03:30:14.808604563 +0000
@@ -314,7 +314,7 @@
 # for creating them.
 $(1):
 	$(if $(quiet),@echo "    [LD] $$@")
-	$(qexec)$$(LD) -dynamiclib $$(LDFLAGS) \
+	$(qexec)$$(LD) -dynamiclib $$(LDFLAGS) -install_name $(PREFIX)/lib/$$@ \
 	    -exported_symbols_list $$(EXPORTS_FILE) \
         -Wl,-headerpad_max_install_names,-compatibility_version,1.0,-current_version,$$(VERSION_MAJOR) \
         -o $$@ \
