diff -ru zip232/unix/unix.c zip232+iPhone/unix/unix.c
--- zip232/unix/unix.c	2006-05-29 22:35:00.000000000 +0000
+++ zip232+iPhone/unix/unix.c	2008-06-18 15:38:50.000000000 +0000
@@ -829,14 +829,7 @@
 #endif /* SGI */
 
 
-/* Define the compile date string */
-#ifdef __DATE__
-#  define COMPILE_DATE " on " __DATE__
-#else
-#  define COMPILE_DATE ""
-#endif
-
-    printf("Compiled with %s for Unix (%s)%s.\n\n",
-           COMPILER_NAME, OS_NAME, COMPILE_DATE);
+    printf("Compiled with %s for Unix (%s).\n\n",
+           COMPILER_NAME, OS_NAME);
 
 } /* end function version_local() */
