diff -Naur tigervnc-1.10.1/CMakeLists.txt tigervnc-1.10.1+iPhone/CMakeLists.txt
--- tigervnc-1.10.1/CMakeLists.txt	2019-12-19 23:02:02.000000000 -0800
+++ tigervnc-1.10.1+iPhone/CMakeLists.txt	2020-01-17 08:52:54.000000000 -0800
@@ -225,7 +225,7 @@
 set(FLTK_SKIP_FORMS TRUE)
 find_package(FLTK)
 
-if(UNIX AND NOT APPLE)
+if(UNIX)
   # No proper handling for extra X11 libs that FLTK might need...
   if(X11_Xft_FOUND)
     # Xft headers include references to fontconfig, so we need
@@ -282,28 +282,12 @@
 
 add_subdirectory(common)
 
-if(WIN32)
-  add_subdirectory(win)
-else()
-  # No interest in building x related parts on Apple
-  if(NOT APPLE)
-    add_subdirectory(unix)
-  endif()
-endif()
+add_subdirectory(unix)
 
 if(ENABLE_NLS)
   add_subdirectory(po)
 endif()
 
-option(BUILD_VIEWER "Build TigerVNC viewer" ON)
-if(BUILD_VIEWER)
-  add_subdirectory(vncviewer)
-  add_subdirectory(media)
-endif()
-
-add_subdirectory(tests)
-
-
 include(cmake/BuildPackages.cmake)
 
 # uninstall
diff -Naur tigervnc-1.10.1/vncviewer/CMakeLists.txt tigervnc-1.10.1+iPhone/vncviewer/CMakeLists.txt
--- tigervnc-1.10.1/vncviewer/CMakeLists.txt	2019-12-19 23:02:02.000000000 -0800
+++ tigervnc-1.10.1+iPhone/vncviewer/CMakeLists.txt	2020-01-17 08:45:24.000000000 -0800
@@ -28,16 +28,12 @@
 
 if(WIN32)
   set(VNCVIEWER_SOURCES ${VNCVIEWER_SOURCES} win32.c)
-elseif(APPLE)
-  set(VNCVIEWER_SOURCES ${VNCVIEWER_SOURCES} cocoa.mm osx_to_qnum.c)
 else()
   set(VNCVIEWER_SOURCES ${VNCVIEWER_SOURCES} xkb_to_qnum.c)
 endif()
 
 if(WIN32)
   set(VNCVIEWER_SOURCES ${VNCVIEWER_SOURCES} Surface_Win32.cxx)
-elseif(APPLE)
-  set(VNCVIEWER_SOURCES ${VNCVIEWER_SOURCES} Surface_OSX.cxx)
 else()
   set(VNCVIEWER_SOURCES ${VNCVIEWER_SOURCES} Surface_X11.cxx)
 endif()
@@ -54,12 +50,6 @@
   target_link_libraries(vncviewer msimg32)
 endif()
 
-if(APPLE)
-  target_link_libraries(vncviewer "-framework Cocoa")
-  target_link_libraries(vncviewer "-framework Carbon")
-  target_link_libraries(vncviewer "-framework IOKit")
-endif()
-
 install(TARGETS vncviewer DESTINATION ${BIN_DIR})
 if(UNIX)
   install(FILES vncviewer.man DESTINATION ${MAN_DIR}/man1 RENAME vncviewer.1)
