diff -ru IOKitTools-76/ioalloccount.tproj/ioalloccount.c IOKitTools-76+iPhone/ioalloccount.tproj/ioalloccount.c
--- IOKitTools-76/ioalloccount.tproj/ioalloccount.c	2003-11-18 19:23:01.000000000 +0000
+++ IOKitTools-76+iPhone/ioalloccount.tproj/ioalloccount.c	2008-03-24 11:14:24.000000000 +0000
@@ -6,6 +6,7 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/IOKitLib.h>
 #include <IOKit/IOKitKeys.h>
+#include <MacTypes.h>
 
 static void printNumber( CFDictionaryRef dict, CFStringRef name )
 {
diff -ru IOKitTools-76/ioclasscount.tproj/ioclasscount.c IOKitTools-76+iPhone/ioclasscount.tproj/ioclasscount.c
--- IOKitTools-76/ioclasscount.tproj/ioclasscount.c	2003-11-06 19:16:54.000000000 +0000
+++ IOKitTools-76+iPhone/ioclasscount.tproj/ioclasscount.c	2008-03-24 11:14:24.000000000 +0000
@@ -8,6 +8,7 @@
 
 #include <IOKit/IOKitLib.h>
 #include <IOKit/IOKitKeys.h>
+#include <MacTypes.h>
 
 static int CompareKey(const void * left, const void * right)
 {
diff -ru IOKitTools-76/ioreg.tproj/ioreg.c IOKitTools-76+iPhone/ioreg.tproj/ioreg.c
--- IOKitTools-76/ioreg.tproj/ioreg.c	2007-01-23 21:29:08.000000000 +0000
+++ IOKitTools-76+iPhone/ioreg.tproj/ioreg.c	2008-03-24 11:14:47.000000000 +0000
@@ -28,6 +28,7 @@
 #include <sys/ioctl.h>                                // (TIOCGWINSZ, ...)
 #include <term.h>                                     // (tputs, ...)
 #include <unistd.h>                                   // (getopt, ...)
+#include <MacTypes.h>
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
