diff -ru libdict-0.9/src/examples/cpptest.cpp libdict-0.9+iPhone/src/examples/cpptest.cpp
--- libdict-0.9/src/examples/cpptest.cpp	2002-03-15 19:03:40.000000000 +0000
+++ libdict-0.9+iPhone/src/examples/cpptest.cpp	2008-01-01 23:33:06.000000000 +0000
@@ -32,6 +32,8 @@
 #include <unistd.h>
 #include "libdict-wrap.hpp"
 
+using namespace std;
+
 int main(int argc, char **argv) {
 	class dictConn dc;
 	char optch;

