diff -ru libdict-0.9/src/examples/Makefile libdict-0.9+iPhone/src/examples/Makefile
--- libdict-0.9/src/examples/Makefile	2002-03-14 17:23:04.000000000 +0000
+++ libdict-0.9+iPhone/src/examples/Makefile	2008-01-01 23:32:52.000000000 +0000
@@ -7,8 +7,8 @@
 all: dictcli cpptest
 
 dictcli:
-	cc -o dictcli dictcli.c ${LIBS} ${CFLAGS} ${CWARN} ${LDFLAGS}
+	$(CC) -o dictcli dictcli.c ${LIBS} ${CFLAGS} ${CWARN} ${LDFLAGS}
 cpptest:
-	c++ -o cpptest cpptest.cpp ${LIBS} ${CFLAGS} ${CWARN} ${LDFLAGS}
+	$(CXX) -o cpptest cpptest.cpp ${LIBS} ${CFLAGS} ${CWARN} ${LDFLAGS}
 clean:
 	rm -f *.o dictcli cpptest *.core
diff -ru libdict-0.9/src/Makefile libdict-0.9+iPhone/src/Makefile
--- libdict-0.9/src/Makefile	2002-03-15 19:10:17.000000000 +0000
+++ libdict-0.9+iPhone/src/Makefile	2008-01-01 23:32:32.000000000 +0000
@@ -18,13 +18,13 @@
 	${CXX} ${CFLAGS} -c libdict-wrap.cpp
 libdict.a: ${OBJ} ${CPPOBJ}
 	rm -f libdict.a
-	ar rc libdict.a *.o
-	ranlib libdict.a
+	$${PKG_TARG}-ar rc libdict.a *.o
+	$${PKG_TARG}-ranlib libdict.a
 clean: clean-example
 	rm -f *.o *.core core 
 superclean: clean
 	rm -f libdict.a
 example:
-	(cd examples; make)
+	(cd examples; $(MAKE))
 clean-example:
-	(cd examples; make clean)
+	(cd examples; $(MAKE) clean)
