diff -ru krb5-1.6.3/src/lib/kadm5/clnt/Makefile.in krb5-1.6.3+iPhone/src/lib/kadm5/clnt/Makefile.in
--- krb5-1.6.3/src/lib/kadm5/clnt/Makefile.in	2007-09-18 04:22:24.000000000 +0000
+++ krb5-1.6.3+iPhone/src/lib/kadm5/clnt/Makefile.in	2008-01-02 01:01:40.000000000 +0000
@@ -5,6 +5,8 @@
 LOCALINCLUDES = -I$(BUILDTOP)/include/kadm5
 DEFS=
 
+MAKE_SHLIB_COMMAND += $(LIBRESOLV)
+
 LIBBASE=kadm5clnt
 LIBMAJOR=5
 LIBMINOR=1
diff -ru krb5-1.6.3/src/lib/kadm5/srv/Makefile.in krb5-1.6.3+iPhone/src/lib/kadm5/srv/Makefile.in
--- krb5-1.6.3/src/lib/kadm5/srv/Makefile.in	2007-09-18 04:22:24.000000000 +0000
+++ krb5-1.6.3+iPhone/src/lib/kadm5/srv/Makefile.in	2008-01-02 01:02:05.000000000 +0000
@@ -8,6 +8,8 @@
 DEFINES = @HESIOD_DEFS@
 DEFS=
 
+MAKE_SHLIB_COMMAND += $(LIBRESOLV)
+
 ##DOSBUILDTOP = ..\..\..
 ##DOSLIBNAME = libkadm5srv.lib
 
diff -ru krb5-1.6.3/src/lib/krb4/Makefile.in krb5-1.6.3+iPhone/src/lib/krb4/Makefile.in
--- krb5-1.6.3/src/lib/krb4/Makefile.in	2007-09-18 04:22:24.000000000 +0000
+++ krb5-1.6.3+iPhone/src/lib/krb4/Makefile.in	2008-01-02 00:58:53.000000000 +0000
@@ -6,6 +6,8 @@
 DEFINES= -DKRB4_USE_KEYTAB
 DEFS=
 
+MAKE_SHLIB_COMMAND += $(LIBRESOLV)
+
 ##DOS##BUILDTOP = ..\..
 ##DOS##LIBNAME=$(OUTPRE)krb4.lib
 ##DOS##OBJFILE=$(OUTPRE)krb4.lst
diff -ru krb5-1.6.3/src/lib/rpc/Makefile.in krb5-1.6.3+iPhone/src/lib/rpc/Makefile.in
--- krb5-1.6.3/src/lib/rpc/Makefile.in	2006-10-05 14:05:40.000000000 +0000
+++ krb5-1.6.3+iPhone/src/lib/rpc/Makefile.in	2008-01-02 01:01:02.000000000 +0000
@@ -5,6 +5,8 @@
 DEFINES = -DGSSAPI_KRB5 -DDEBUG_GSSAPI=0 -DGSSRPC__IMPL
 DEFS=
 
+MAKE_SHLIB_COMMAND += $(LIBRESOLV)
+
 SUBDIRS=unit-test
 
 ##DOSBUILDTOP = ..\..
