diff -ru krb5-1.6.3/src/config/pre.in krb5-1.6.3+iPhone/src/config/pre.in
--- krb5-1.6.3/src/config/pre.in	2006-10-15 18:19:28.000000000 +0000
+++ krb5-1.6.3+iPhone/src/config/pre.in	2008-01-02 01:03:37.000000000 +0000
@@ -158,7 +158,7 @@
 # PTHREAD_CFLAGS set by configure, not included in CFLAGS so that we
 #		don't pull the pthreads library into shared libraries
 ALL_CFLAGS = $(DEFS) $(DEFINES) $(KRB_INCLUDES) $(LOCALINCLUDES) \
-	-DKRB5_DEPRECATED=1 -DKRB5_PRIVATE=1 \
+	-DKRB5_DEPRECATED=1 -DKRB5_PRIVATE=1 -fno-common \
 	$(CPPFLAGS) $(CFLAGS) $(PTHREAD_CFLAGS)
 
 CFLAGS = @CFLAGS@
