diff -ur cvs-1.12.13/src/Makefile.am cvs-1.12.13+iPhone/src/Makefile.am
--- cvs-1.12.13/src/Makefile.am	2018-09-04 10:32:01.000000000 -1000
+++ cvs-1.12.13+iPhone/src/Makefile.am	2018-09-04 10:32:57.000000000 -1000
@@ -110,7 +110,6 @@
 	../diff/libdiff.a \
 	../lib/libcvs.a \
 	$(ZLIB_LIBS)
-cvs_LDFLAGS = -Wl,--as-needed
 cvs_LDADD = $(cvs_client_objects) \
 	../diff/libdiff.a \
 	../lib/libcvs.a \
