diff -ur tcsh-TCSH6_20_00/config/mach tcsh-TCSH6_20_00+iPhone/config/mach
--- tcsh-TCSH6_20_00/config/bsd4.4	2016-11-24 05:26:18.000000000 -1000
+++ tcsh-TCSH6_20_00+iPhone/config/bsd4.4	2019-03-29 18:15:08.000000000 -1000
@@ -95,4 +95,6 @@
 # define _TERMIOS_H_
 #endif /* __NeXT31__  && !POSIX */
 
+#define HASHBANG
+
 #endif /* _h_config */
diff -ur tcsh-TCSH6_20_00/sh.exec.c tcsh-TCSH6_20_00+iPhone/sh.exec.c
--- tcsh-TCSH6_20_00/sh.exec.c	2016-11-24 05:26:18.000000000 -1000
+++ tcsh-TCSH6_20_00+iPhone/sh.exec.c	2019-03-29 18:14:16.000000000 -1000
@@ -401,6 +401,7 @@
     switch (errno) {
 
     case ENOEXEC:
+    case EPERM:
 #ifdef WINNT_NATIVE
 		nt_feed_to_cmd(f,t);
 #endif /* WINNT_NATIVE */
