diff -ru libnet/aclocal.m4 libnet+iPhone/aclocal.m4
--- libnet/aclocal.m4	2004-03-01 20:26:11.000000000 +0000
+++ libnet+iPhone/aclocal.m4	2009-01-20 21:44:39.000000000 +0000
@@ -294,6 +294,8 @@
 AC_DEFUN(AC_LIBNET_ENDIAN_CHECK,
     [AC_MSG_CHECKING(machine endianess)
 
+    if test ! ${ac_cv_libnet_endianess+@}; then
+
     cat > conftest.c << EOF
 #       include <stdio.h>
 #       include <stdlib.h>
@@ -346,6 +348,8 @@
         fi
         rm -f conftest* core core.conftest
 
+        fi
+
         AC_MSG_RESULT($ac_cv_libnet_endianess)
 
         if test $ac_cv_libnet_endianess = big ; then
