--- subversion-1.10.2/subversion/libsvn_subr/opt.c	2015-10-04 07:15:07.000000000 -1000
+++ subversion-1.10.2+iPhone/subversion/libsvn_subr/opt.c	2018-08-17 16:07:19.000000000 -1000
@@ -1118,12 +1118,7 @@
   if (quiet)
     return svn_cmdline_printf(pool, "%s\n", SVN_VER_NUMBER);
 
-  SVN_ERR(svn_cmdline_printf(pool, _("%s, version %s\n"
-                                     "   compiled %s, %s on %s\n\n"),
-                             pgm_name, SVN_VERSION,
-                             svn_version_ext_build_date(info),
-                             svn_version_ext_build_time(info),
-                             svn_version_ext_build_host(info)));
+  SVN_ERR(svn_cmdline_printf(pool, "%s, version %s\n", pgm_name, SVN_VERSION));
   SVN_ERR(svn_cmdline_printf(pool, "%s\n", svn_version_ext_copyright(info)));
 
   if (footer)
