diff -ur cmake-3.14.2/Source/cmGlobalXCodeGenerator.cxx cmake-3.14.2+iPhone/Source/cmGlobalXCodeGenerator.cxx
--- cmake-3.14.2/Source/cmGlobalXCodeGenerator.cxx	2019-04-12 12:10:08.000000000 +0000
+++ cmake-3.14.2+iPhone/Source/cmGlobalXCodeGenerator.cxx	2019-04-17 20:39:56.609722294 +0000
@@ -36,7 +36,7 @@
 
 struct cmLinkImplementation;
 
-#if defined(CMAKE_BUILD_WITH_CMAKE) && defined(__APPLE__)
+#if 0 // defined(CMAKE_BUILD_WITH_CMAKE) && defined(__APPLE__)
 #  define HAVE_APPLICATION_SERVICES
 #  include <ApplicationServices/ApplicationServices.h>
 #endif
