diff -ru slirp-1.0.16/src/configure.in slirp-1.0.16+iPhone/src/configure.in
--- slirp-1.0.16/src/configure.in	2004-09-01 09:58:50.000000000 +0000
+++ slirp-1.0.16+iPhone/src/configure.in	2008-08-18 19:38:00.000000000 +0000
@@ -47,9 +47,6 @@
 fi
 AC_PROG_GCC_TRADITIONAL
 dnl AC_PROG_INSTALL
-if test x$cross_compiling = xyes; then
-	eval $CROSS_ERROR
-fi
 dnl - Check for perl 5
 AC_PATH_PROG(PERL, perl, no)
 if test x$PERL = xno; then
