diff -ru weechat-0.2.6/configure.in weechat-0.2.6+iPhone/configure.in
--- weechat-0.2.6/configure.in	2007-09-05 14:14:00.000000000 +0000
+++ weechat-0.2.6+iPhone/configure.in	2008-08-18 18:02:35.000000000 +0000
@@ -605,8 +597,8 @@
 *** libgnutls was not found. You may want to get it from ftp://ftp.gnutls.org/pub/gnutls/
 *** WeeChat will be built without GnuTLS support.]]))
     if test "x$found_gnutls" = "xyes" ; then
-        GNUTLS_CFLAGS=`libgnutls-config --cflags`
-        GNUTLS_LFLAGS=`libgnutls-config --libs`
+        GNUTLS_CFLAGS=$LIBGNUTLS_CFLAGS
+        GNUTLS_LFLAGS=$LIBGNUTLS_LIBS
         AC_SUBST(GNUTLS_CFLAGS)
         AC_SUBST(GNUTLS_LFLAGS)
         AC_DEFINE(HAVE_GNUTLS)
