diff -r -u weechat-0.2.6/Makefile.in weechat-0.2.6+iPhone/Makefile.in
--- weechat-0.2.6/Makefile.in	2007-09-05 14:16:27.000000000 +0000
+++ weechat-0.2.6+iPhone/Makefile.in	2008-07-31 03:03:29.000000000 +0000
@@ -276,7 +276,7 @@
              debian/weechat-curses.menu debian/NEWS debian/weechat.xpm \
              weechat.spec weechat_icon_32.png
 
-ACLOCAL_AMFLAGS = -I m4
+ACLOCAL_AMFLAGS = 
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
