--- libxml2-2.6.16/parser.c	2004-11-09 15:48:18.000000000 +0000
+++ libxml2-2.6.16+iPhone/parser.c	2007-08-10 01:39:38.000000000 +0000
@@ -9253,6 +9276,8 @@
 		(!ctxt->disableSAX))
 		ctxt->sax->endElementNs(ctxt->userData, name, prefix, URI);
+		if (ctxt->nsNr - nsNr > 0)
+			nsPop(ctxt, ctxt->nsNr - nsNr);
 #ifdef LIBXML_SAX1_ENABLED
 	} else {
 	    if ((ctxt->sax != NULL) && (ctxt->sax->endElement != NULL) &&
