diff -ru ncftp-3.2.1/ncftp/cmds.c ncftp-3.2.1+iPhone/ncftp/cmds.c
--- ncftp-3.2.1/ncftp/cmds.c	2007-07-27 03:11:04.000000000 +0000
+++ ncftp-3.2.1+iPhone/ncftp/cmds.c	2008-06-19 04:46:12.000000000 +0000
@@ -4536,9 +4536,6 @@
 	(void) printf("Archived at:      ftp://ftp.NcFTP.com/ncftp/\n");
 #endif
 	(void) printf("Library Version:  %s\n", gLibNcFTPVersion + 5);
-#ifdef __DATE__
-	(void) printf("Compile Date:     %s\n", __DATE__);
-#endif
 	if (gOS[0] != '\0')
 		(void) printf("Platform:         %s\n", gOS);
 }	/* VersionCmd */
