diff -ru ncftp-3.2.1/configure ncftp-3.2.1+iPhone/configure
--- ncftp-3.2.1/configure	2007-08-01 05:39:16.000000000 +0000
+++ ncftp-3.2.1+iPhone/configure	2008-03-24 09:22:22.000000000 +0000
@@ -8387,10 +8387,13 @@
 	case "$CC" in
 		*${machine}*)
 			f=`echo "$CC" | sed 's|/[^/]*$||'`
+			AR=$(which "${machine}-ar") ||
 			if [ -x "$f/${machine}-ar" ] ; then
 				AR="$f/${machine}-ar"
 			elif [ -x "$f/ar" ] ; then
 				AR="$f/ar"
+			else
+				AR="ar"
 			fi
 			;;
 	esac
