diff -ur inetutils-1.9.4/Makefile.am inetutils-1.9.4+iPhone/Makefile.am
--- inetutils-1.9.4/Makefile.am	2015-06-08 21:45:35.000000000 -1000
+++ inetutils-1.9.4+iPhone/Makefile.am	2018-08-17 15:13:33.000000000 -1000
@@ -25,7 +25,6 @@
 SUBDIRS = lib \
 	libinetutils libtelnet libicmp libls \
 	src telnet telnetd ftp ftpd talk talkd whois ping ifconfig \
-	doc man \
 	tests
 
 DISTCLEANFILES = pathdefs.make paths.defs \
diff -ur inetutils-1.9.4/Makefile.in inetutils-1.9.4+iPhone/Makefile.in
--- inetutils-1.9.4/Makefile.in	2015-06-08 21:48:08.000000000 -1000
+++ inetutils-1.9.4+iPhone/Makefile.in	2018-08-17 15:13:50.000000000 -1000
@@ -1416,7 +1416,6 @@
 SUBDIRS = lib \
 	libinetutils libtelnet libicmp libls \
 	src telnet telnetd ftp ftpd talk talkd whois ping ifconfig \
-	doc man \
 	tests
 
 DISTCLEANFILES = pathdefs.make paths.defs \
