diff -ur inetutils-1.9.4/src/rlogind.c inetutils-1.9.4+iPhone/src/rlogind.c
--- inetutils-1.9.4/src/rlogind.c	2015-05-12 02:13:25.000000000 -1000
+++ inetutils-1.9.4+iPhone/src/rlogind.c	2018-08-17 15:08:20.000000000 -1000
@@ -1172,7 +1172,7 @@
   struct auth_data auth_data;
   char addrstr[INET6_ADDRSTRLEN];
   const char *reply;
-  int true;
+  int _true;
   char c;
   int authenticated;
   pid_t pid;
@@ -1204,16 +1204,16 @@
 	  ? ntohs (((struct sockaddr_in6 *) &auth_data.from)->sin6_port)
 	  : ntohs (((struct sockaddr_in *) &auth_data.from)->sin_port));
 
-  true = 1;
+  _true = 1;
   if (keepalive
-      && setsockopt (infd, SOL_SOCKET, SO_KEEPALIVE, &true, sizeof true) < 0)
+      && setsockopt (infd, SOL_SOCKET, SO_KEEPALIVE, &_true, sizeof _true) < 0)
     syslog (LOG_WARNING, "setsockopt (SO_KEEPALIVE): %m");
 
 #if defined IP_TOS && defined IPPROTO_IP && defined IPTOS_LOWDELAY
-  true = IPTOS_LOWDELAY;
+  _true = IPTOS_LOWDELAY;
   if (auth_data.from.ss_family == AF_INET &&
       setsockopt (infd, IPPROTO_IP, IP_TOS,
-		  (char *) &true, sizeof true) < 0)
+		  (char *) &_true, sizeof _true) < 0)
     syslog (LOG_WARNING, "setsockopt (IP_TOS): %m");
 #endif
 
@@ -1260,10 +1260,10 @@
     }
 
   /* Parent */
-  true = 1;
-  IF_NOT_ENCRYPT (ioctl (infd, FIONBIO, &true));
-  ioctl (master, FIONBIO, &true);
-  ioctl (master, TIOCPKT, &true);
+  _true = 1;
+  IF_NOT_ENCRYPT (ioctl (infd, FIONBIO, &_true));
+  ioctl (master, FIONBIO, &_true);
+  ioctl (master, TIOCPKT, &_true);
   netf = infd;			/* Needed for cleanup() */
   setsig (SIGCHLD, cleanup);
   protocol (infd, master, &auth_data);
