diff -ur ruby-2.5.3/configure.ac ruby-2.5.3+iPhone/configure.ac
--- ruby-2.5.3/configure.ac	2018-01-03 08:12:16.000000000 -1000
+++ ruby-2.5.3+iPhone/configure.ac	2018-11-14 00:02:11.000000000 -1000
@@ -465,13 +465,6 @@
 		 CFLAGS="$save_CFLAGS"
 	 ])])
 
-AS_CASE(["$target_os"], [darwin*], [
-if libtool 2>&1 | grep no_warning_for_no_symbols > /dev/null; then
-  ac_cv_prog_ac_ct_RANLIB=:
-  ac_cv_prog_ac_ct_AR='libtool -static'
-  rb_cv_arflags='-no_warning_for_no_symbols -o'
-fi
-])
 AC_CHECK_TOOL(RANLIB, ranlib, :)
 AC_CHECK_TOOL(AR, ar)
 AS_IF([test -z "$AR"], [
