diff -ru rlwrap-0.29/src/ptytty.c rlwrap-0.29+iPhone/src/ptytty.c
--- rlwrap-0.29/src/ptytty.c	2007-09-16 10:49:21.000000000 +0000
+++ rlwrap-0.29+iPhone/src/ptytty.c	2008-06-03 08:53:10.000000000 +0000
@@ -41,7 +41,7 @@
 #ifdef HAVE_SYS_IOCTL_H
 # include <sys/ioctl.h>
 #endif
-#if defined(PTYS_ARE_PTMX) && !defined(__CYGWIN32__)
+#if defined(PTYS_ARE_PTMX) && !defined(__CYGWIN32__) && !defined(__APPLE__)
 # include <sys/stropts.h>	/* for I_PUSH */
 #endif
 
