diff -ur openssh-7.7p1/session.c openssh-7.7p1+iPhone/session.c
--- openssh-7.7p1/session.c	2018-04-01 19:38:28.000000000 -1000
+++ openssh-7.7p1+iPhone/session.c	2018-07-24 12:09:29.000000000 -1000
@@ -141,7 +141,7 @@
 extern void destroy_sensitive_data(void);
 extern Buffer loginmsg;
 extern struct sshauthopt *auth_opts;
-char *tun_fwd_ifnames; /* serverloop.c */
+extern char *tun_fwd_ifnames; /* serverloop.c */
 
 /* original command from peer. */
 const char *original_command = NULL;
