Only in ld64-274.2+iPhone: InputFiles.o
diff -ur ld64-274.2/src/ld/InputFiles.cpp ld64-274.2+iPhone/src/ld/InputFiles.cpp
--- ld64-274.2/src/ld/InputFiles.cpp	2016-07-01 15:46:14.000000000 -1000
+++ ld64-274.2+iPhone/src/ld/InputFiles.cpp	2018-09-05 11:18:34.000000000 -1000
@@ -881,6 +881,8 @@
 	opts.setArchitecture(CPU_TYPE_X86_64, CPU_SUBTYPE_X86_64_ALL, Options::kPlatformOSX);
 #elif __arm__
 	opts.setArchitecture(CPU_TYPE_ARM, CPU_SUBTYPE_ARM_V6, Options::kPlatformOSX);
+#elif __arm64__
+	opts.setArchitecture(CPU_TYPE_ARM64, CPU_SUBTYPE_ARM_V8, Options::kPlatformOSX);
 #else
 	#error unknown default architecture
 #endif
