diff -ur ld64-274.2/src/ld/parsers/textstub_dylib_file.cpp ld64-274.2+iPhone/src/ld/parsers/textstub_dylib_file.cpp
--- ld64-274.2/src/ld/parsers/textstub_dylib_file.cpp	2016-07-26 11:15:31.000000000 -1000
+++ ld64-274.2+iPhone/src/ld/parsers/textstub_dylib_file.cpp	2018-09-05 12:09:16.000000000 -1000
@@ -29,7 +29,7 @@
 #include <vector>
 
 #include "Architectures.hpp"
-#include "bitcode.hpp"
+#include "Bitcode.hpp"
 #include "MachOFileAbstraction.hpp"
 #include "MachOTrie.hpp"
 #include "generic_dylib_file.hpp"
