diff -ur ld64-274.2/src/ld/Options.cpp ld64-274.2+iPhone/src/ld/Options.cpp
--- ld64-274.2/src/ld/Options.cpp	2017-01-20 12:24:09.000000000 -1000
+++ ld64-274.2+iPhone/src/ld/Options.cpp	2018-09-05 18:36:45.000000000 -1000
@@ -45,6 +45,8 @@
 #include "MachOFileAbstraction.hpp"
 #include "Snapshot.h"
 
+const char *ldVersionString = "@(#)PROGRAM:ld  PROJECT:ld64-" VERSION "\n";
+
 
 // from FunctionNameDemangle.h
 extern "C" size_t fnd_get_demangled_name(const char *mangledName, char *outputBuffer, size_t length);
@@ -3946,7 +3948,6 @@
 			addStandardLibraryDirectories = false;
 		else if ( strcmp(argv[i], "-v") == 0 ) {
 			fVerbose = true;
-			extern const char ldVersionString[];
 			fprintf(stderr, "%s", ldVersionString);
 			fprintf(stderr, "configured to support archs: %s\n", ALL_SUPPORTED_ARCHS);
 			 // if only -v specified, exit cleanly
@@ -5864,7 +5865,6 @@
 		uint8_t version = depLinkerVersion;
 		if ( write(fDependencyFileDescriptor, &version, 1) == -1 )
 			throwf("write() to -dependency_info failed, errno=%d", errno);
-		extern const char ldVersionString[];
 		if ( write(fDependencyFileDescriptor, ldVersionString, strlen(ldVersionString)+1) == -1 )
 			throwf("write() to -dependency_info failed, errno=%d", errno);
 	}
