diff -ru gdb-1119/Makefile gdb-1119+iPhone/Makefile
--- gdb-1119/Makefile	2009-04-15 23:42:09.000000000 +0000
+++ gdb-1119+iPhone/Makefile	2009-07-07 11:02:05.000000000 +0000
@@ -293,8 +293,8 @@
 		fi;\
 	done;
 	(cd $(SYMROOT)/$(LIBEXEC_GDB_DIR)/ ; dsymutil gdb-arm-apple-darwin)
-	strip -S -o $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-arm-apple-darwin \
-		$(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-arm-apple-darwin
+	cp $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-arm-apple-darwin \
+		$(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-arm-apple-darwin
 	chown root:wheel $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-arm-apple-darwin
 	chmod 755 $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-arm-apple-darwin
 	mkdir -p ${DSTROOT}/usr/bin
@@ -372,7 +372,7 @@
 			fi; \
 		done; \
                 (cd $(SYMROOT)/$(LIBEXEC_GDB_DIR)/ ; dsymutil gdb-$${target_arch_vendor_os}); \
-		strip -S -o $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target_arch_vendor_os} "$${curr_symroot_output_file}"; \
+		cp $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target_arch_vendor_os} "$${curr_symroot_output_file}"; \
 	done; \
 	mkdir -p ${DSTROOT}/usr/bin; \
 	sed -e 's/version=.*/version=$(GDB_VERSION)-$(GDB_RC_VERSION)/' \
@@ -690,8 +690,8 @@
 			-output $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
 		dsymutil -o $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}.dSYM \
                          $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
-	 	strip -S -o $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target} \
-			$(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
+	 	cp $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target} \
+			$(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
 		cp $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target} $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
 	done
 
@@ -705,8 +705,8 @@
 	set -e; for target in $(CANONICAL_ARCHS); do \
 		dsymutil -o $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}.dSYM \
                          $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
-	 	strip -S -o $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target} \
-			$(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
+	 	cp $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target} \
+			$(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
 		cp $(DSTROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target} \
                    $(SYMROOT)/$(LIBEXEC_GDB_DIR)/gdb-$${target}; \
 		if echo $${target} | egrep '^[^-]*-apple-darwin' > /dev/null; then \
@@ -730,7 +729,7 @@
 		instname=`echo $${i} | sed -e 's/\\-new//'`; \
 		lipo -create $(patsubst %,$(OBJROOT)/%/binutils/$${i},$(NATIVE_TARGETS)) \
 			-output $(SYMROOT)/$(LIBEXEC_BINUTILS_DIR)/$${instname}; \
-		strip -S -o $(DSTROOT)/$(LIBEXEC_BINUTILS_DIR)/$${instname} $(SYMROOT)/$(LIBEXEC_BINUTILS_DIR)/$${instname}; \
+		cp $(SYMROOT)/$(LIBEXEC_BINUTILS_DIR)/$${instname} $(DSTROOT)/$(LIBEXEC_BINUTILS_DIR)/$${instname}; \
 	done
 
 install-chmod-macosx:
