diff -r -u lsof-33/lsof/Configure lsof-33+iPhone/lsof/Configure
--- lsof-33/lsof/Configure	2007-05-26 01:18:17.000000000 +0000
+++ lsof-33+iPhone/lsof/Configure	2008-04-18 09:29:18.000000000 +0000
@@ -4835,7 +4835,7 @@
 if test "X$LSOF_LIB_NO" = "X"	# {
 then
   echo "" >> $LSOF_MKFC
-  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB -llsof $LSOF_CFGL" >> $LSOF_MKFC
+  echo "CFGL=	$LSOF_FCFGL -L./$LSOF_LIB lib/*.o $LSOF_CFGL" >> $LSOF_MKFC
 fi	# }
 echo "" >> $LSOF_MKFC
 if test "X$LSOF_DEBUG" = "X"	# {
