diff -Naur libgpg-error-1.36/src/mkheader.c libgpg-error-1.36+iPhone/src/mkheader.c
--- libgpg-error-1.36/src/mkheader.c	2019-03-19 00:04:48.000000000 -0700
+++ libgpg-error-1.36+iPhone/src/mkheader.c	2019-11-16 22:37:35.000000000 -0800
@@ -110,6 +110,7 @@
 
     {"powerpc-unknown-linux-gnuspe", "powerpc-unknown-linux-gnu" },
 
+    {"%TARGET%", "aarch64-apple-darwin" },
     {"arm-unknown-linux-gnueabihf",  "arm-unknown-linux-gnueabi" },
     {"armv7-unknown-linux-gnueabihf"  },
     {"armv7a-unknown-linux-gnueabihf" },
