diff -ur libgpg-error-1.32/src/mkheader.c libgpg-error-1.32+iPhone/src/mkheader.c
--- libgpg-error-1.32/src/mkheader.c	2018-03-13 22:56:20.000000000 -1000
+++ libgpg-error-1.32+iPhone/src/mkheader.c	2018-08-01 15:13:26.000000000 -1000
@@ -90,6 +90,9 @@
     {"powerpc-unknown-linux-gnuspe", "powerpc-unknown-linux-gnu" },
 
     {"arm-unknown-linux-gnueabihf",  "arm-unknown-linux-gnueabi" },
+    {"arm-apple-darwin11",  "arm-apple-darwin" },
+    {"aarch64-apple-darwin14",  "aarch64-apple-darwin" },
+    {"aarch64-apple-darwin17",  "aarch64-apple-darwin" },
     {"armv7-unknown-linux-gnueabihf"  },
     {"armv5-unknown-linux-musleabi"   },
     {"armv6-unknown-linux-musleabihf" },
