diff -ur mobileterminal-364/Sources/MobileTerminal.h mobileterminal-364+iPhone/Sources/MobileTerminal.h
--- mobileterminal-364/Sources/MobileTerminal.h	2009-04-11 17:48:49.000000000 -1000
+++ mobileterminal-364+iPhone/Sources/MobileTerminal.h	2018-09-02 13:21:13.000000000 -1000
@@ -7,6 +7,8 @@
 #import "Constants.h"
 #import "Log.h"
 
+typedef struct __GSEvent* GSEventRef;
+
 
 @class PTYTextView;
 @class ShellKeyboard;
