diff -ru mobileterminal-364/Sources/UI/MainViewController.m mobileterminal-364+iPhone/Sources/UI/MainViewController.m
--- mobileterminal-364/Sources/UI/MainViewController.m	2009-04-26 07:54:34.000000000 +0000
+++ mobileterminal-364+iPhone/Sources/UI/MainViewController.m	2009-08-14 12:09:20.000000000 +0000
@@ -265,6 +265,8 @@
 
 - (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)orientation
 {
+    bool yes = targetOrientation_ == UIDeviceOrientationUnknown;
+
     int currentOrientation = [self interfaceOrientation];
     if (currentOrientation && currentOrientation != orientation) {
         targetOrientation_ = orientation;
@@ -280,6 +282,9 @@
         [keyboardView setAlpha:0.0f];
         [self.activeTextView.tiledView setAlpha:0.0f];
         [UIView commitAnimations];
+    } else if (yes) {
+        targetOrientation_ = orientation;
+        return YES;
     }
 
     return NO;
