diff -ur coreutils-8.30/Makefile.am coreutils-8.30+iPhone/Makefile.am
--- coreutils-8.30/Makefile.am	2018-05-13 18:20:24.000000000 -1000
+++ coreutils-8.30+iPhone/Makefile.am	2018-08-02 22:58:28.000000000 -1000
@@ -211,5 +211,4 @@
 include $(top_srcdir)/lib/local.mk
 include $(top_srcdir)/src/local.mk
 include $(top_srcdir)/doc/local.mk
-include $(top_srcdir)/man/local.mk
 include $(top_srcdir)/tests/local.mk
