diff -ur x264-snapshot-20190402-2245/Makefile x264-snapshot-20190402-2245+iPhone/Makefile
--- x264-snapshot-20190402-2245/Makefile	2019-04-02 10:45:02.000000000 -1000
+++ x264-snapshot-20190402-2245+iPhone/Makefile	2019-04-02 21:24:44.000000000 -1000
@@ -240,7 +240,7 @@
 
 $(LIBX264): $(GENERATED) .depend $(OBJS) $(OBJASM)
 	rm -f $(LIBX264)
-	$(AR)$@ $(OBJS) $(OBJASM)
+	$(AR) -r $@ $(OBJS) $(OBJASM)
 	$(if $(RANLIB), $(RANLIB) $@)
 
 $(SONAME): $(GENERATED) .depend $(OBJS) $(OBJASM) $(OBJSO)
