diff -ru rdesktop-1.5.0/Makefile.in rdesktop-1.5.0+iPhone/Makefile.in
--- rdesktop-1.5.0/Makefile.in	2006-04-09 18:27:50.000000000 +0000
+++ rdesktop-1.5.0+iPhone/Makefile.in	2008-04-04 20:02:01.000000000 +0000
@@ -57,7 +57,6 @@
 installbin: rdesktop
 	mkdir -p $(DESTDIR)$(bindir)
 	$(INSTALL) rdesktop $(DESTDIR)$(bindir)
-	$(STRIP) $(DESTDIR)$(bindir)/rdesktop
 	chmod 755 $(DESTDIR)$(bindir)/rdesktop
 
 .PHONY: installman
