diff -ru jocstrap-1.0.2549/jocstrap/makefile jocstrap-1.0.2549+iPhone/jocstrap/makefile
--- jocstrap-1.0.2549/jocstrap/makefile	2008-08-26 11:08:53.000000000 +0000
+++ jocstrap-1.0.2549+iPhone/jocstrap/makefile	2008-08-26 11:13:02.000000000 +0000
@@ -43,7 +43,6 @@
     frameworks += MessageUI
     frameworks += AddressBookUI
     frameworks += CoreTelephony
-    frameworks += CoreLocation
     frameworks += MusicLibrary
     frameworks += CoreLocation
     main += HelloJava
