diff -ru vpnc-0.5.1/Makefile vpnc-0.5.1+iPhone/Makefile
--- vpnc-0.5.1/Makefile	2007-09-06 20:05:15.000000000 +0000
+++ vpnc-0.5.1+iPhone/Makefile	2008-02-27 11:47:52.000000000 +0000
@@ -46,8 +46,8 @@
 
 # Comment this in to obtain a binary with certificate support which is
 # GPL incompliant though.
-#OPENSSL_GPL_VIOLATION = -DOPENSSL_GPL_VIOLATION
-#OPENSSLLIBS = -lcrypto
+OPENSSL_GPL_VIOLATION = -DOPENSSL_GPL_VIOLATION
+OPENSSLLIBS = -lcrypto
 
 CC=gcc
 CFLAGS ?= -O3 -g
