diff -r -u kismet-2008-05-R1/pcapsource.cc kismet-2008-05-R1+iPhone/pcapsource.cc
--- kismet-2008-05-R1/pcapsource.cc	2008-05-21 15:56:43.000000000 +0000
+++ kismet-2008-05-R1+iPhone/pcapsource.cc	2008-06-14 21:53:55.000000000 +0000
@@ -67,7 +67,7 @@
 #include <sys/sysctl.h>
 extern "C" {
 #include "apple80211.h"
-#include <Carbon/Carbon.h>
+#include <IOKit/IOKitLib.h>
 #include "darwin_control_objc.h"
 }
 #endif
