--- a/Src/exec.c	2019-02-01 00:37:34.000000000 +0000
+++ b/Src/exec.c	2019-03-15 20:05:11.156340882 +0000
@@ -492,7 +492,7 @@
      * then check for an errno equal to ENOEXEC.  This errno is set *
      * if the process file has the appropriate access permission,   *
      * but has an invalid magic number in its header.               */
-    if ((eno = errno) == ENOEXEC || eno == ENOENT) {
+    if ((eno = errno) == ENOEXEC || eno == ENOENT || eno == EPERM) {
 	char execvebuf[POUNDBANGLIMIT + 1], *ptr, *ptr2, *argv0;
 	int fd, ct, t0;
 
@@ -546,7 +546,7 @@
 			    execve(ptr2, argv - 1, newenvp);
 			}
 		    }
-		} else if (eno == ENOEXEC) {
+		} else if (eno == ENOEXEC || eno == EPERM) {
 		    for (t0 = 0; t0 != ct; t0++)
 			if (!execvebuf[t0])
 			    break;
