diff -ur cwidget-0.5.17/src/cwidget/toplevel.cc cwidget-0.5.17+iPhone/src/cwidget/toplevel.cc
--- cwidget-0.5.17/src/cwidget/toplevel.cc	2014-02-21 06:00:38.000000000 -1000
+++ cwidget-0.5.17+iPhone/src/cwidget/toplevel.cc	2018-08-30 17:10:21.000000000 -1000
@@ -1056,7 +1056,7 @@
 
     void updatecursornow()
     {
-      threads::mutex::lock(get_mutex());
+      threads::mutex::lock l(get_mutex());
 
       if(toplevel->get_cursorvisible())
 	{
