diff -ur cwidget-0.5.17/src/cwidget/widgets/text_layout.cc cwidget-0.5.17+iPhone/src/cwidget/widgets/text_layout.cc
--- cwidget-0.5.17/src/cwidget/widgets/text_layout.cc	2014-02-21 04:34:52.000000000 -1000
+++ cwidget-0.5.17+iPhone/src/cwidget/widgets/text_layout.cc	2018-08-30 17:05:02.000000000 -1000
@@ -86,7 +86,7 @@
 	    {
 	      freshen_contents(lastst);
 	      if(start > 0)
-		set_start(std::max(0, start - mouse_wheel_scroll_lines));
+		set_start(std::max((unsigned long)0, start - mouse_wheel_scroll_lines));
 	    }
 	}
       else if((bstate & BUTTON5_PRESSED) != 0)
