diff -ur cwidget-0.5.17/src/cwidget/curses++.h cwidget-0.5.17+iPhone/src/cwidget/curses++.h
--- cwidget-0.5.17/src/cwidget/curses++.h	2014-02-21 04:34:52.000000000 -1000
+++ cwidget-0.5.17+iPhone/src/cwidget/curses++.h	2018-08-30 16:30:44.000000000 -1000
@@ -58,9 +58,7 @@
      */
     attr_t attrs;
 
-    wchtype()
-    {
-    }
+    wchtype() = default;
 
     wchtype(const wchar_t &_ch, const attr_t &_attrs)
       :ch(_ch), attrs(_attrs)
