diff -ur cwidget-0.5.17/src/cwidget/generic/threads/threads.h cwidget-0.5.17+iPhone/src/cwidget/generic/threads/threads.h
--- cwidget-0.5.17/src/cwidget/generic/threads/threads.h	2014-02-21 04:34:52.000000000 -1000
+++ cwidget-0.5.17+iPhone/src/cwidget/generic/threads/threads.h	2018-08-30 16:56:11.000000000 -1000
@@ -25,6 +25,7 @@
 #ifndef THREADS_H
 #define THREADS_H
 
+#include <pthread.h>
 #include <errno.h>
 #include <cwidget/generic/util/exception.h>
 
