diff -ur libvpx-1.8.0/libs.mk libvpx-1.8.0+iPhone/libs.mk
--- libvpx-1.8.0/libs.mk	2019-02-04 07:02:33.000000000 -1000
+++ libvpx-1.8.0+iPhone/libs.mk	2019-04-03 14:50:11.000000000 -1000
@@ -145,7 +145,6 @@
 endif
 CODEC_EXPORTS-yes += vpx/exports_com
 CODEC_EXPORTS-$(CONFIG_ENCODERS) += vpx/exports_enc
-CODEC_EXPORTS-$(CONFIG_VP9_ENCODER) += vpx/exports_spatial_svc
 CODEC_EXPORTS-$(CONFIG_DECODERS) += vpx/exports_dec
 
 INSTALL-LIBS-yes += include/vpx/vpx_codec.h
