diff -ur system_cmds-790.30.1/arch.tproj/arch.c system_cmds-790.30.1+iPhone/arch.tproj/arch.c
--- system_cmds-790.30.1/arch.tproj/arch.c	2018-02-03 12:02:36.000000000 -1000
+++ system_cmds-790.30.1+iPhone/arch.tproj/arch.c	2018-07-31 16:15:42.000000000 -1000
@@ -78,6 +78,8 @@
     {"x86_64", CPU_TYPE_X86_64},
 #elif defined(__arm__)
     {"arm", CPU_TYPE_ARM},
+#elif defined(__arm64__)
+    {"arm", CPU_TYPE_ARM64},
 #else
 #error "Unsupported architecture"
 #endif
@@ -102,6 +104,9 @@
 #elif defined(__arm__)
 #define NATIVE_32	"arm"
 #define NATIVE_64	NULL
+#elif defined(__arm64__)
+#define NATIVE_32	NULL
+#define NATIVE_64	"arm64"
 #else
 #error "Unsupported architecture"
 #endif
