diff -ru lynx2-8-6/src/LYMain.c lynx2-8-6+iPhone/src/LYMain.c
--- lynx2-8-6/src/LYMain.c	2006-09-19 00:28:28.000000000 +0000
+++ lynx2-8-6+iPhone/src/LYMain.c	2008-01-31 08:06:40.000000000 +0000
@@ -1,3 +1,5 @@
+#include <mach-o/nlist.h>
+
 #include <HTUtils.h>
 #include <HTTP.h>
 #include <HTParse.h>
@@ -961,6 +963,15 @@
 int main(int argc,
 	 char **argv)
 {
+#if !defined(__ENVIRONMENT_ASPEN_VERSION_MIN_REQUIRED__) || __ENVIRONMENT_ASPEN_VERSION_MIN_REQUIRED__ < 10200
+    struct nlist nl[2];
+    memset(nl, 0, sizeof(nl));
+    nl[0].n_un.n_name = (char *) "_useMDNSResponder";
+    nlist("/usr/lib/libc.dylib", nl);
+    if (nl[0].n_type != N_UNDF)
+        *(int *) nl[0].n_value = 0;
+#endif
+
     int i;			/* indexing variable */
     int status = 0;		/* exit status */
     char *temp = NULL;
