diff -ru lynx2-8-6/src/LYMain.c lynx2-8-6+iPhone/src/LYMain.c
--- lynx2-8-6/src/LYMain.c	2006-09-19 00:28:28.000000000 +0000
+++ lynx2-8-6+iPhone/src/LYMain.c	2008-06-18 01:31:49.000000000 +0000
@@ -3144,39 +3144,6 @@
     printf("%s\n", result);
     free(result);
 
-#ifndef __DATE__
-#define __DATE__ ""
-#endif
-#ifndef __TIME__
-#define __TIME__ ""
-#endif
-
-/*
- * SYSTEM_NAME is set by the configure script.  Show build date/time for other
- * systems, according to predefined compiler symbols.
- */
-#ifdef SYSTEM_NAME
-    printf(gettext("Built on %s %s %s\n"), SYSTEM_NAME, __DATE__, __TIME__);
-#else
-#ifdef __CYGWIN__
-    printf("Compiled by CYGWIN (%s %s).\n", __DATE__, __TIME__);
-#else
-#ifdef __BORLANDC__
-    printf("Compiled by Borland C++ (%s %s).\n", __DATE__, __TIME__);
-#else
-#ifdef _MSC_VER
-    printf("Compiled by Microsoft Visual C++ (%s %s).\n", __DATE__, __TIME__);
-#else
-#ifdef __DJGPP__
-    printf("Compiled by DJGPP (%s %s).\n", __DATE__, __TIME__);
-#else
-    printf("Compiled at (%s %s).\n", __DATE__, __TIME__);
-#endif /* __DJGPP__ */
-#endif /* _MSC_VER */
-#endif /* __BORLANDC__ */
-#endif /* __CYGWIN__ */
-#endif
-
     puts("");
     puts(gettext("Copyrights held by the University of Kansas, CERN, and other contributors."));
     puts(gettext("Distributed under the GNU General Public License."));
