diff -ru top-39/top.h top-39+iPhone/top.h
--- top-39/top.h	2007-05-21 20:43:56.000000000 +0000
+++ top-39+iPhone/top.h	2009-05-12 10:34:55.000000000 +0000
@@ -174,7 +174,7 @@
 #define FORMAT_TIME_HHMMSS   'w'
 
 #define VALID_FORMATS "abcdefghijklmnopqrstuvw"
-#define FORMAT_ESCAPE        '\'
+#define FORMAT_ESCAPE        '\\'
 #define FORMAT_LEFT          '^'
 #define FORMAT_RIGHT         '$'
 #define FORMAT_DELTA         '-'
