diff -ru top-39/top.h top-39+iPhone/top.h
--- top-39/top.h	2007-05-21 20:43:56.000000000 +0000
+++ top-39+iPhone/top.h	2009-05-12 00:51:45.000000000 +0000
@@ -180,10 +180,10 @@
 #define FORMAT_DELTA         '-'
 #define VALID_ESCAPES        "\\^$-"
 
-#define LEGEND_CN  "  PID COMMAND      %CPU   TIME   #TH #PRTS #MREGS RPRVT  RSHRD  RSIZE  VSIZE"
-#define FORMAT_CN  "$aaaa ^bbbbbbbbb $cccc% $wwwwwww $ee $ffff-$ggggg $hhhh- $iiii- $jjjj- $llll-"
-#define LEGEND_CNW "  PID COMMAND      %CPU   TIME   #TH #PRTS(delta) #MREGS RPRVT(delta) RSHRD(delta) RSIZE(delta) VSIZE(delta)"
-#define FORMAT_CNW "$aaaa ^bbbbbbbbb $cccc% $wwwwwww $ee $ffff------- $ggggg $hhhh------- $iiii------- $jjjj------- $llll-------"
+#define LEGEND_CN  "  PID COMMAND      %CPU   TIME   #TH #PRTS #MREGS  RPRVT  RSHRD  RSIZE  VSIZE"
+#define FORMAT_CN  "$aaaa ^bbbbbbbbb $cccc% $wwwwwww $ee $ffff-$ggggg $hhhhh- $iiii- $jjjj- $llll-"
+#define LEGEND_CNW "  PID COMMAND      %CPU   TIME   #TH #PRTS(delta) #MREGS  RPRVT(delta) RSHRD(delta) RSIZE(delta) VSIZE(delta)"
+#define FORMAT_CNW "$aaaa ^bbbbbbbbb $cccc% $wwwwwww $ee $ffff------- $ggggg $hhhhh------- $iiii------- $jjjj------- $llll-------"
 #define LEGEND_CA  "  PID COMMAND      %CPU   TIME    FAULTS   PAGEINS COW_FAULTS MSGS_SENT  MSGS_RCVD  BSDSYSCALL MACHSYSCALL CSWITCH"
 #define FORMAT_CA  "$aaaa ^bbbbbbbbb $cccc% $wwwwwww ^oooooo  ^ppppppp ^qqqqqqqqq ^rrrrrrrr  ^ssssssss  ^ttttttttt ^uuuuuuuuuu ^vvvvvv"
 #define LEGEND_CD  "  PID COMMAND      %CPU   TIME   FAULTS PGINS/COWS MSENT/MRCVD    BSD/MACH      CSW"
@@ -191,18 +191,18 @@
 #define LEGEND_CE LEGEND_CA
 #define FORMAT_CE FORMAT_CA
 
-#define LEGEND_XCN  "  PID   UID  REG RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT    TIME  %CPU COMMAND"
-#define FORMAT_XCN  "$aaaa $mmmm $ggg $hhhh- $iiii- $jjjj- $kkkk- $llll-$ee $ff-$dddddd $cccc ^bbbbbbbbbbbbbbb"
-#define LEGEND_XCNW "  PID   UID  REG RPRVT( delta) RSHRD( delta) RSIZE( delta) VPRVT( delta) VSIZE( delta) TH PRT(delta)   TIME  %CPU COMMAND"
-#define FORMAT_XCNW "$aaaa $mmmm $ggg $hhhh-------- $iiii-------- $jjjj-------- $kkkk-------- $llll--------$ee $ff-------$dddddd $cccc ^bbbbbbbbbbbbbbb"
-#define LEGEND_XTCN  "  PID USERNAME  REG  RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT    TIME  %CPU COMMAND"
-#define FORMAT_XTCN  "$aaaa ^nnnnnnnn $ggg $hhhh- $iiii- $jjjj- $kkkk- $llll-$ee $ff-$dddddd $cccc ^bbbbbbbbbbbbbbb"
-#define LEGEND_XTCNW "  PID USERNAME  REG  RPRVT( delta) RSHRD( delta) RSIZE( delta) VPRVT( delta) VSIZE( delta) TH PRT(delta)   TIME  %CPU COMMAND"
-#define FORMAT_XTCNW "$aaaa ^nnnnnnnn $ggg $hhhh-------- $iiii-------- $jjjj-------- $kkkk-------- $llll--------$ee $ff-------$dddddd $cccc ^bbbbbbbbbbbbbbb"
-#define LEGEND_XCA "  PID   UID  REG RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT      FAULTS    PAGEINS COW_FAULTS  MSGS_SENT  MSGS_RCVD   BSYSCALL   MSYSCALL    CSWITCH   TIME  %CPU COMMAND"
-#define FORMAT_XCA "$aaaa $mmmm $ggg $hhhh  $iiii $jjjjj  $kkkk  $llll $ee $ff  $ooooooooo  $pppppppp $qqqqqqqqq  $rrrrrrrr  $ssssssss   $ttttttt   $uuuuuuu    $vvvvvv $ddddd$ccccc $bbbbbbbbbbbbbbbb"
-#define LEGEND_XTCA "  PID USERNAME  REG RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT      FAULTS    PAGEINS COW_FAULTS  MSGS_SENT  MSGS_RCVD   BSYSCALL   MSYSCALL    CSWITCH   TIME  %CPU COMMAND"
-#define FORMAT_XTCA "$aaaa ^nnnnnnnn $ggg $hhhh  $iiii $jjjjj  $kkkk  $llll $ee $ff  $ooooooooo  $ppppppp $qqqqqqqqq  $rrrrrrrr  $ssssssss   $ttttttt   $uuuuuuu    $vvvvvv $ddddd$ccccc $bbbbbbbbbbbbbbbb"
+#define LEGEND_XCN  "  PID   UID  REG  RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT    TIME  %CPU COMMAND"
+#define FORMAT_XCN  "$aaaa $mmmm $ggg $hhhhh- $iiii- $jjjj- $kkkk- $llll-$ee $ff-$dddddd $cccc ^bbbbbbbbbbbbbbb"
+#define LEGEND_XCNW "  PID   UID  REG  RPRVT( delta) RSHRD( delta) RSIZE( delta) VPRVT( delta) VSIZE( delta) TH PRT(delta)   TIME  %CPU COMMAND"
+#define FORMAT_XCNW "$aaaa $mmmm $ggg $hhhhh-------- $iiii-------- $jjjj-------- $kkkk-------- $llll--------$ee $ff-------$dddddd $cccc ^bbbbbbbbbbbbbbb"
+#define LEGEND_XTCN  "  PID USERNAME  REG   RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT    TIME  %CPU COMMAND"
+#define FORMAT_XTCN  "$aaaa ^nnnnnnnn $ggg $hhhhh- $iiii- $jjjj- $kkkk- $llll-$ee $ff-$dddddd $cccc ^bbbbbbbbbbbbbbb"
+#define LEGEND_XTCNW "  PID USERNAME  REG   RPRVT( delta) RSHRD( delta) RSIZE( delta) VPRVT( delta) VSIZE( delta) TH PRT(delta)   TIME  %CPU COMMAND"
+#define FORMAT_XTCNW "$aaaa ^nnnnnnnn $ggg $hhhhh-------- $iiii-------- $jjjj-------- $kkkk-------- $llll--------$ee $ff-------$dddddd $cccc ^bbbbbbbbbbbbbbb"
+#define LEGEND_XCA "  PID   UID  REG  RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT      FAULTS    PAGEINS COW_FAULTS  MSGS_SENT  MSGS_RCVD   BSYSCALL   MSYSCALL    CSWITCH   TIME  %CPU COMMAND"
+#define FORMAT_XCA "$aaaa $mmmm $ggg $hhhhh  $iiii $jjjjj  $kkkk  $llll $ee $ff  $ooooooooo  $pppppppp $qqqqqqqqq  $rrrrrrrr  $ssssssss   $ttttttt   $uuuuuuu    $vvvvvv $ddddd$ccccc $bbbbbbbbbbbbbbbb"
+#define LEGEND_XTCA "  PID USERNAME  REG  RPRVT  RSHRD  RSIZE  VPRVT  VSIZE  TH PRT      FAULTS    PAGEINS COW_FAULTS  MSGS_SENT  MSGS_RCVD   BSYSCALL   MSYSCALL    CSWITCH   TIME  %CPU COMMAND"
+#define FORMAT_XTCA "$aaaa ^nnnnnnnn $ggg $hhhhh  $iiii $jjjjj  $kkkk  $llll $ee $ff  $ooooooooo  $ppppppp $qqqqqqqqq  $rrrrrrrr  $ssssssss   $ttttttt   $uuuuuuu    $vvvvvv $ddddd$ccccc $bbbbbbbbbbbbbbbb"
 #define LEGEND_XCD LEGEND_XCA
 #define FORMAT_XCD FORMAT_XCA
 #define LEGEND_XCE LEGEND_XCA
