diff -u top-37/Makefile top-37+iPhone/Makefile
--- top-37/Makefile	2006-06-19 22:22:33.000000000 +0000
+++ top-37+iPhone/Makefile	2007-12-27 12:03:07.000000000 +0000
@@ -77,7 +77,7 @@
 
 installbin : $(SYMROOT)/$(BIN)
 	$(INSTALL) -d $(DSTROOT)/usr/bin
-	$(INSTALL) -s -m 4755 $< $(DSTROOT)/usr/bin
+	$(INSTALL) -m 4755 $< $(DSTROOT)/usr/bin
 
 installman :
 	$(INSTALL) -d $(DSTROOT)/usr/share/man/man1
