diff -ru apt-0.7.25.3/apt-pkg/deb/deblistparser.cc apt-0.7.25.3+iPhone/apt-pkg/deb/deblistparser.cc
--- apt-0.7.25.3/apt-pkg/deb/deblistparser.cc	2010-02-01 19:44:40.000000000 +0000
+++ apt-0.7.25.3+iPhone/apt-pkg/deb/deblistparser.cc	2011-05-27 05:46:33.000000000 +0000
@@ -605,6 +605,9 @@
       if (stringcmp(Start,Stop,"all") == 0)
 	 return true;
 
+      if (stringcmp(Start,Stop,"cydia") == 0)
+	 return true;
+
       iOffset = Tags.Offset();
    }   
    return false;
