diff -ru apt-0.7.25.3/apt-pkg/contrib/macros.h apt-0.7.25.3+iPhone/apt-pkg/contrib/macros.h
--- apt-0.7.25.3/apt-pkg/contrib/macros.h	2010-02-22 18:11:17.000000000 +0000
+++ apt-0.7.25.3+iPhone/apt-pkg/contrib/macros.h	2010-02-01 19:44:40.000000000 +0000
@@ -57,7 +57,9 @@
 // some nice optional GNUC features
 #if __GNUC__ >= 3
         #define __must_check    __attribute__ ((warn_unused_result))
+#ifndef __deprecated
         #define __deprecated    __attribute__ ((deprecated))
+#endif
         /* likely() and unlikely() can be used to mark boolean expressions
            as (not) likely true which will help the compiler to optimise */
         #define likely(x)       __builtin_expect (!!(x), 1)
