diff -ru apt-0.7.20.2/apt-pkg/makefile apt-0.7.20.2+iPhone/apt-pkg/makefile
--- apt-0.7.20.2/apt-pkg/makefile	2009-02-07 15:09:35.000000000 +0000
+++ apt-0.7.20.2+iPhone/apt-pkg/makefile	2009-04-15 19:36:09.000000000 +0000
@@ -15,7 +15,7 @@
 LIBRARY=apt-pkg
 MAJOR=$(LIBAPTPKG_MAJOR)
 MINOR=$(LIBAPTPKG_RELEASE)
-SLIBS=$(PTHREADLIB) $(INTLLIBS) -lutil -ldl
+SLIBS=$(PTHREADLIB) $(INTLLIBS) -lutil -ldl $(LIBICONV)
 APT_DOMAIN:=libapt-pkg$(LIBAPTPKG_MAJOR)
 
 # Source code for the contributed non-core things
diff -ru apt-0.7.20.2/buildlib/environment.mak.in apt-0.7.20.2+iPhone/buildlib/environment.mak.in
--- apt-0.7.20.2/buildlib/environment.mak.in	2009-02-07 15:09:35.000000000 +0000
+++ apt-0.7.20.2+iPhone/buildlib/environment.mak.in	2009-04-15 19:36:23.000000000 +0000
@@ -55,6 +55,7 @@
 PYTHONINCLUDE = @PYTHONINCLUDE@
 BDBLIB = @BDBLIB@
 INTLLIBS = @INTLLIBS@
+LIBICONV = @LIBICONV@
 
 # Shim Headerfile control
 HAVE_C9X = @HAVE_C9X@
