diff -ru apt-0.7.25.3/apt-pkg/pkgcache.cc apt-0.7.25.3+iPhone/apt-pkg/pkgcache.cc
--- apt-0.7.25.3/apt-pkg/pkgcache.cc	2010-02-01 19:44:40.000000000 +0000
+++ apt-0.7.25.3+iPhone/apt-pkg/pkgcache.cc	2011-03-11 01:33:44.000000000 +0000
@@ -185,7 +185,7 @@
    Package *Pkg = PkgP + HeaderP->HashTable[Hash(Name)];
    for (; Pkg != PkgP; Pkg = PkgP + Pkg->NextPackage)
    {
-      if (Pkg->Name != 0 && StrP[Pkg->Name] == Name[0] &&
+      if (Pkg->Name != 0 &&
 	  stringcasecmp(Name,StrP + Pkg->Name) == 0)
 	 return PkgIterator(*this,Pkg);
    }
