diff -ru apt-0.7.20.2/apt-inst/makefile apt-0.7.20.2+iPhone/apt-inst/makefile
--- apt-0.7.20.2/apt-inst/makefile	2009-02-07 15:09:35.000000000 +0000
+++ apt-0.7.20.2+iPhone/apt-inst/makefile	2009-04-16 01:54:17.000000000 +0000
@@ -14,7 +14,7 @@
 LIBEXT=$(GLIBC_VER)$(LIBSTDCPP_VER)
 MAJOR=1.1
 MINOR=0
-SLIBS=$(PTHREADLIB) -lapt-pkg
+SLIBS=$(PTHREADLIB) -lapt-pkg $(INTLLIBS)
 APT_DOMAIN:=libapt-inst$(MAJOR)
 
 # Source code for the contributed non-core things
