diff -ru apt-0.7.20.2/apt-pkg/deb/debindexfile.h apt-0.7.20.2+iPhone/apt-pkg/deb/debindexfile.h
--- apt-0.7.20.2/apt-pkg/deb/debindexfile.h	2009-02-07 15:09:35.000000000 +0000
+++ apt-0.7.20.2+iPhone/apt-pkg/deb/debindexfile.h	2009-04-19 04:12:51.000000000 +0000
@@ -49,9 +49,10 @@
 
    string Info(const char *Type) const;
    string IndexFile(const char *Type) const;
-   string IndexURI(const char *Type) const;
    
    public:
+
+   string IndexURI(const char *Type) const;
    
    virtual const Type *GetType() const;
 
