diff -ru libffi-3.0.10/src/arm/ffi.c libffi-3.0.10+iPhone/src/arm/ffi.c
--- libffi-3.0.10/src/arm/ffi.c	2011-08-23 14:28:04.000000000 +0000
+++ libffi-3.0.10+iPhone/src/arm/ffi.c	2012-03-18 23:34:32.000000000 +0000
@@ -32,6 +32,10 @@
 
 #include <stdlib.h>
 
+#ifdef __APPLE__
+#define __clear_cache(a, b)
+#endif
+
 /* Forward declares. */
 static int vfp_type_p (ffi_type *);
 static void layout_vfp_args (ffi_cif *);
