diff -ru ipkg-0.99.163/libbb/libbb.h ipkg-0.99.163+iPhone/libbb/libbb.h
--- ipkg-0.99.163/libbb/libbb.h	2006-02-06 08:13:02.000000000 +0000
+++ ipkg-0.99.163+iPhone/libbb/libbb.h	2009-06-12 00:06:09.000000000 +0000
@@ -50,8 +50,6 @@
 #include "dmalloc.h"
 #endif
 
-#include <features.h>
-
 #if (__GNU_LIBRARY__ < 5) && (!defined __dietlibc__)
 /* libc5 doesn't define socklen_t */
 typedef unsigned int socklen_t;
