diff -ur aptitude-0.8.10/src/generic/util/parsers.h aptitude-0.8.10+iPhone/src/generic/util/parsers.h
--- aptitude-0.8.10/src/generic/util/parsers.h	2017-08-19 07:59:07.000000000 -1000
+++ aptitude-0.8.10+iPhone/src/generic/util/parsers.h	2018-08-31 09:36:06.000000000 -1000
@@ -72,12 +72,6 @@
 
 #include <errno.h>
 
-namespace std
-{
-  template<typename T, typename Alloc>
-  class deque;
-}
-
 namespace parsers
 {
   /** \defgroup Parser combinators in C++
