diff -ur aptitude-0.8.10/src/pkg_columnizer.cc aptitude-0.8.10+iPhone/src/pkg_columnizer.cc
--- aptitude-0.8.10/src/pkg_columnizer.cc	2017-08-19 07:59:07.000000000 -1000
+++ aptitude-0.8.10+iPhone/src/pkg_columnizer.cc	2018-08-31 15:13:18.000000000 -1000
@@ -70,8 +70,8 @@
   {30, false, false},   // origin
   {10, true, true},     // archive
   {9, false, false},    // sizechange
-  {strlen(PACKAGE), false, false},  // progname
-  {strlen(VERSION), false, false},  // progver
+  {(unsigned int)strlen(PACKAGE), false, false},  // progname
+  {(unsigned int)strlen(VERSION), false, false},  // progver
   {12, false, false},   // brokencount
   {16, false, false},   // diskusage, example (max): "Disk: -2,587 kB "
   {22, false, false},   // downloadsize, example (max): " DL: 2,586 kB/2,587 kB"
@@ -396,7 +396,8 @@
     case diskusage:
       {
 	size_t bufsize = 256;
-	char buf[bufsize] = "";
+	char buf[bufsize];
+	buf[0] = 0;
 	if (apt_cache_file && ((*apt_cache_file)->UsrSize() != 0))
 	  {
 	    char sign = ((*apt_cache_file)->UsrSize() > 0) ? '+' : '-';
