diff -ur aptitude-0.8.10/src/generic/util/util.cc aptitude-0.8.10+iPhone/src/generic/util/util.cc
--- aptitude-0.8.10/src/generic/util/util.cc	2017-08-19 07:59:07.000000000 -1000
+++ aptitude-0.8.10+iPhone/src/generic/util/util.cc	2018-08-31 09:24:23.000000000 -1000
@@ -242,7 +242,7 @@
 	  // We need to copy "result", not "buf", because some
 	  // versions of strerror_r can return a static string and
 	  // leave "buf" alone.
-	  string rval(result);
+	  string rval(buf);
 	  delete[] buf;
 	  return rval;
 	}
