diff -ru lighttpd-1.4.18/src/server.c lighttpd-1.4.18+iPhone/src/server.c
--- lighttpd-1.4.18/src/server.c	2007-09-05 10:39:56.000000000 +0000
+++ lighttpd-1.4.18+iPhone/src/server.c	2008-06-18 03:52:58.000000000 +0000
@@ -322,8 +322,7 @@
 # define TEXT_SSL
 #endif
 	char *b = PACKAGE_NAME "-" PACKAGE_VERSION TEXT_SSL \
-" - a light and fast webserver\n" \
-"Build-Date: " __DATE__ " " __TIME__ "\n";
+" - a light and fast webserver\n";
 ;
 #undef TEXT_SSL
 	write(STDOUT_FILENO, b, strlen(b));
@@ -458,7 +457,7 @@
 #else
 # define TEXT_SSL
 #endif
-	char *b = PACKAGE_NAME "-" PACKAGE_VERSION TEXT_SSL " ("__DATE__ " " __TIME__ ")" \
+	char *b = PACKAGE_NAME "-" PACKAGE_VERSION TEXT_SSL \
 " - a light and fast webserver\n" \
 "usage:\n" \
 " -f <name>  filename of the config-file\n" \
