diff -Naur Python-3.7.4/setup.py Python-3.7.4+iPhone/setup.py
--- Python-3.7.4/setup.py	2019-07-08 11:03:50.000000000 -0700
+++ Python-3.7.4+iPhone/setup.py	2019-10-17 09:48:02.000000000 -0700
@@ -114,7 +114,7 @@
     if m is not None:
         MACOS_SDK_ROOT = m.group(1)
     else:
-        MACOS_SDK_ROOT = '/'
+        MACOS_SDK_ROOT = sysroot = os.environ['PKG_ROOT'] or '/'
         cc = sysconfig.get_config_var('CC')
         tmpfile = '/tmp/setup_sdk_root.%d' % os.getpid()
         try:
@@ -650,11 +650,10 @@
             # Add the sysroot paths. 'sysroot' is a compiler option used to
             # set the logical path of the standard system headers and
             # libraries.
-            lib_dirs = (self.compiler.library_dirs +
-                        sysroot_paths(('LDFLAGS', 'CC'), system_lib_dirs))
-            inc_dirs = (self.compiler.include_dirs +
-                        sysroot_paths(('CPPFLAGS', 'CFLAGS', 'CC'),
-                                      system_include_dirs))
+            lib_dirs = ([os.getenv('PKG_ROOT') + '/usr/lib'] +
+                        os.getenv('LIBRARY_PATH').split(':'))
+            inc_dirs = ([os.getenv('PKG_ROOT') + '/usr/include'] +
+                        os.getenv('CPATH').split(':'))
         exts = []
         missing = []
 
