diff -ur Python-2.7.15/Modules/getbuildinfo.c Python-2.7.15+iPhone/Modules/getbuildinfo.c
--- Python-2.7.15/Modules/getbuildinfo.c	2018-04-29 12:47:33.000000000 -1000
+++ Python-2.7.15+iPhone/Modules/getbuildinfo.c	2018-09-04 16:04:45.000000000 -1000
@@ -5,20 +5,12 @@
 #endif
 
 #ifndef DATE
-#ifdef __DATE__
-#define DATE __DATE__
-#else
 #define DATE "xx/xx/xx"
 #endif
-#endif
 
 #ifndef TIME
-#ifdef __TIME__
-#define TIME __TIME__
-#else
 #define TIME "xx:xx:xx"
 #endif
-#endif
 
 /* XXX Only unix build process has been tested */
 #ifndef GITVERSION
