diff -ur cpio-2.12/gnu/progname.c cpio-2.12+iPhone/gnu/progname.c
--- cpio-2.12/gnu/progname.c	2015-08-31 09:58:43.000000000 -1000
+++ cpio-2.12+iPhone/gnu/progname.c	2018-08-14 11:30:38.000000000 -1000
@@ -28,10 +28,6 @@
 #include <string.h>
 
 
-/* String containing name the program is called with.
-   To be initialized by main().  */
-const char *program_name = NULL;
-
 /* Set program_name, based on argv[0].
    argv0 must be a string allocated with indefinite extent, and must not be
    modified after this call.  */
