diff -ur mc-4.8.21/lib/global.h mc-4.8.21+iPhone/lib/global.h
--- mc-4.8.21/lib/global.h	2018-05-27 09:56:21.000000000 -1000
+++ mc-4.8.21+iPhone/lib/global.h	2018-08-14 10:59:08.000000000 -1000
@@ -24,6 +24,12 @@
 /* for O_* macros */
 #include <fcntl.h>
 
+#ifdef __APPLE__
+#define st_atim st_atimespec
+#define st_ctim st_ctimespec
+#define st_mtim st_mtimespec
+#endif
+
 /* for sig_atomic_t */
 #include <signal.h>
 
