diff -ur mc-4.8.21/Makefile.am mc-4.8.21+iPhone/Makefile.am
--- mc-4.8.21/Makefile.am	2017-02-24 10:25:57.000000000 -1000
+++ mc-4.8.21+iPhone/Makefile.am	2018-08-14 10:44:39.000000000 -1000
@@ -1,7 +1,7 @@
 ## Process this file with automake to create Makefile.in.
 AUTOMAKE_OPTIONS = 1.5
 
-SUBDIRS = intl po lib src doc contrib misc
+SUBDIRS = intl po lib src contrib misc
 
 if HAVE_TESTS
     SUBDIRS += tests
diff -ur mc-4.8.21/Makefile.in mc-4.8.21+iPhone/Makefile.in
--- mc-4.8.21/Makefile.in	2018-06-02 22:57:29.000000000 -1000
+++ mc-4.8.21+iPhone/Makefile.in	2018-08-14 10:44:26.000000000 -1000
@@ -508,7 +508,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.5
-SUBDIRS = intl po lib src doc contrib misc $(am__append_1)
+SUBDIRS = intl po lib src contrib misc $(am__append_1)
 EXTRA_DIST = 
 dist_noinst_SCRIPTS = build-glib2.sh
 dist_noinst_HEADERS = $(top_srcdir)/version.h
