diff -ur indent-2.2.10/Makefile.am indent-2.2.10+iPhone/Makefile.am
--- indent-2.2.10/Makefile.am	2008-03-11 08:50:42.000000000 -1000
+++ indent-2.2.10+iPhone/Makefile.am	2018-09-04 09:52:21.000000000 -1000
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 AUTOMAKE_OPTIONS = no-texinfo.tex
 
-SUBDIRS = intl src doc po man
+SUBDIRS = intl src po
 
 BUILT_SOURCES = 
 
@@ -19,18 +19,6 @@
 
 DISTCLEANFILES=config/config.cache config/config.log config.h
 
-ps:
-	cd doc && $(MAKE) @PACKAGE@.ps
-
-pdf:
-	cd doc && $(MAKE) @PACKAGE@.pdf
-
-html:
-	cd doc && $(MAKE) html
-
-html-split:
-	cd doc && $(MAKE) html-split
-
 # --------------- Maintainer's Section
 
 MAINTAINERCLEANFILES = intl/* aclocal.m4 Makefile.in config.h.in stamp-h1 stamp-h.in configure \
diff -ur indent-2.2.10/configure.ac indent-2.2.10+iPhone/configure.ac
--- indent-2.2.10/configure.ac	2008-03-11 08:50:42.000000000 -1000
+++ indent-2.2.10+iPhone/configure.ac	2018-09-04 09:55:34.000000000 -1000
@@ -63,8 +63,6 @@
    po/Makefile.in
    Makefile
    src/Makefile
-   doc/Makefile
-   man/Makefile
 ],
 [touch Makefile.in])
 
