diff -ur gmp-6.1.2/mpn/arm64/invert_limb.asm gmp-6.1.2+iPhone/mpn/arm64/invert_limb.asm
--- gmp-6.1.2/mpn/arm64/invert_limb.asm	2016-12-16 05:45:27.000000000 -1000
+++ gmp-6.1.2+iPhone/mpn/arm64/invert_limb.asm	2018-08-04 23:39:34.000000000 -1000
@@ -41,9 +41,9 @@
 ASM_START()
 PROLOGUE(mpn_invert_limb)
 	lsr	x2, x0, #54
-	adrp	x1, approx_tab
+	adrp    x1, approx_tab@PAGE
 	and	x2, x2, #0x1fe
-	add	x1, x1, :lo12:approx_tab
+	add     x1, x1, approx_tab@PAGEOFF
 	ldrh	w3, [x1,x2]
 	lsr	x4, x0, #24
 	add	x4, x4, #1
