diff -ru netatalk-2.0.4/macros/db3-check.m4 netatalk-2.0.4+iPhone/macros/db3-check.m4
--- netatalk-2.0.4/macros/db3-check.m4	2009-03-29 07:23:24.000000000 +0000
+++ netatalk-2.0.4+iPhone/macros/db3-check.m4	2009-06-19 22:11:24.000000000 +0000
@@ -33,7 +33,8 @@
     for lib in $atalk_cv_bdb_try_libs ; do
         LIBS="$lib $savedlibs"
         AC_MSG_CHECKING([Berkeley DB library ($lib)])
-        AC_TRY_RUN([
+        AC_LINK_IFELSE([
+            #include "confdefs.h"
             #include <stdio.h>
             #include <db.h>
             int main(void) {
