diff -ru netatalk-2.0.3/bin/adv1tov2/adv1tov2.c netatalk-2.0.3+iPhone/bin/adv1tov2/adv1tov2.c
--- netatalk-2.0.3/bin/adv1tov2/adv1tov2.c	2005-02-10 01:23:05.000000000 +0000
+++ netatalk-2.0.3+iPhone/bin/adv1tov2/adv1tov2.c	2008-07-11 01:10:01.000000000 +0000
@@ -8,6 +8,8 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 
+#include <atalk/util.h>
+
 #include <atalk/adouble.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -25,8 +27,6 @@
 #include <errno.h>
 #include <string.h>
 
-#include <atalk/util.h>
-
 #if AD_VERSION == AD_VERSION2
 #if 0
 /* translate characters */
