diff -ru netatalk-2.0.3/etc/afpd/auth.c netatalk-2.0.3+iPhone/etc/afpd/auth.c
--- netatalk-2.0.3/etc/afpd/auth.c	2005-03-11 15:36:58.000000000 +0000
+++ netatalk-2.0.3+iPhone/etc/afpd/auth.c	2008-10-11 05:57:29.000000000 +0000
@@ -223,11 +223,6 @@
     set_processname("afpd");
     syslog_setup(log_debug, logtype_default, logoption_ndelay|logoption_pid, logfacility_daemon);
 
-    if ( pwd->pw_uid == 0 ) {	/* don't allow root login */
-        LOG(log_error, logtype_afpd, "login: root login denied!" );
-        return AFPERR_NOTAUTH;
-    }
-
     LOG(log_info, logtype_afpd, "login %s (uid %d, gid %d) %s", pwd->pw_name,
         pwd->pw_uid, pwd->pw_gid , afp_versions[afp_version_index].av_name);
 
