diff -ru raop_play-0.5.1/configure raop_play-0.5.1+iPhone/configure
--- raop_play-0.5.1/configure	2005-12-16 14:17:03.000000000 +0000
+++ raop_play-0.5.1+iPhone/configure	2008-12-06 03:45:45.000000000 +0000
@@ -3373,163 +3373,6 @@
 done
 
 
-echo "$as_me:$LINENO: checking for fl_beep in -lfltk" >&5
-echo $ECHO_N "checking for fl_beep in -lfltk... $ECHO_C" >&6
-if test "${ac_cv_lib_fltk_fl_beep+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lfltk  $LIBS"
-cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-/* Override any gcc2 internal prototype to avoid an error.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-/* We use char because int might match the return type of a gcc2
-   builtin and then its argument prototype would still apply.  */
-char dummy(); extern char fl_beep (int a=0);
-int
-main ()
-{
-fl_beep ();
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext conftest$ac_exeext
-if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
-  (eval $ac_link) 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } &&
-	 { ac_try='test -z "$ac_cxx_werror_flag"			 || test ! -s conftest.err'
-  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; } &&
-	 { ac_try='test -s conftest$ac_exeext'
-  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; }; then
-  ac_cv_lib_fltk_fl_beep=yes
-else
-  echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-ac_cv_lib_fltk_fl_beep=no
-fi
-rm -f conftest.err conftest.$ac_objext \
-      conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_fltk_fl_beep" >&5
-echo "${ECHO_T}$ac_cv_lib_fltk_fl_beep" >&6
-if test $ac_cv_lib_fltk_fl_beep = yes; then
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBFLTK 1
-_ACEOF
-
-  LIBS="-lfltk $LIBS"
-
-else
-
-		    echo "Error! You need libfltk."
-		    exit -1
-fi
-
-
-if test -n "$ac_tool_prefix"; then
-  # Extract the first word of "${ac_tool_prefix}fluid", so it can be a program name with args.
-set dummy ${ac_tool_prefix}fluid; ac_word=$2
-echo "$as_me:$LINENO: checking for $ac_word" >&5
-echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6
-if test "${ac_cv_prog_FLUID+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  if test -n "$FLUID"; then
-  ac_cv_prog_FLUID="$FLUID" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-  for ac_exec_ext in '' $ac_executable_extensions; do
-  if $as_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_FLUID="${ac_tool_prefix}fluid"
-    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-done
-
-fi
-fi
-FLUID=$ac_cv_prog_FLUID
-if test -n "$FLUID"; then
-  echo "$as_me:$LINENO: result: $FLUID" >&5
-echo "${ECHO_T}$FLUID" >&6
-else
-  echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6
-fi
-
-fi
-if test -z "$ac_cv_prog_FLUID"; then
-  ac_ct_FLUID=$FLUID
-  # Extract the first word of "fluid", so it can be a program name with args.
-set dummy fluid; ac_word=$2
-echo "$as_me:$LINENO: checking for $ac_word" >&5
-echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6
-if test "${ac_cv_prog_ac_ct_FLUID+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  if test -n "$ac_ct_FLUID"; then
-  ac_cv_prog_ac_ct_FLUID="$ac_ct_FLUID" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-  for ac_exec_ext in '' $ac_executable_extensions; do
-  if $as_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_ac_ct_FLUID="fluid"
-    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-done
-
-  test -z "$ac_cv_prog_ac_ct_FLUID" && ac_cv_prog_ac_ct_FLUID=":"
-fi
-fi
-ac_ct_FLUID=$ac_cv_prog_ac_ct_FLUID
-if test -n "$ac_ct_FLUID"; then
-  echo "$as_me:$LINENO: result: $ac_ct_FLUID" >&5
-echo "${ECHO_T}$ac_ct_FLUID" >&6
-else
-  echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6
-fi
-
-  FLUID=$ac_ct_FLUID
-else
-  FLUID="$ac_cv_prog_FLUID"
-fi
-
 if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}install", so it can be a program name with args.
 set dummy ${ac_tool_prefix}install; ac_word=$2
@@ -6123,7 +5966,7 @@
 done
 
 
-                                                  ac_config_files="$ac_config_files Makefile aexcl/Makefile aexcl/ipod/Makefile raop_play/Makefile rendezvous/Makefile"
+                                                  ac_config_files="$ac_config_files Makefile raop_play/Makefile rendezvous/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -6650,8 +6493,6 @@
   case "$ac_config_target" in
   # Handling of arguments.
   "Makefile" ) CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-  "aexcl/Makefile" ) CONFIG_FILES="$CONFIG_FILES aexcl/Makefile" ;;
-  "aexcl/ipod/Makefile" ) CONFIG_FILES="$CONFIG_FILES aexcl/ipod/Makefile" ;;
   "raop_play/Makefile" ) CONFIG_FILES="$CONFIG_FILES raop_play/Makefile" ;;
   "rendezvous/Makefile" ) CONFIG_FILES="$CONFIG_FILES rendezvous/Makefile" ;;
   "config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
diff -ru raop_play-0.5.1/Makefile.in raop_play-0.5.1+iPhone/Makefile.in
--- raop_play-0.5.1/Makefile.in	2005-12-16 14:16:59.000000000 +0000
+++ raop_play-0.5.1+iPhone/Makefile.in	2008-12-06 01:26:35.000000000 +0000
@@ -1,4 +1,4 @@
-SUBDIRS = rendezvous raop_play aexcl
+SUBDIRS = rendezvous raop_play
 
 all:
 	for i in $(SUBDIRS); do make -C $$i; done
