diff -ru raop_play-0.5.1/raop_play/Makefile.in raop_play-0.5.1+iPhone/raop_play/Makefile.in
--- raop_play-0.5.1/raop_play/Makefile.in	2005-12-16 14:17:00.000000000 +0000
+++ raop_play-0.5.1+iPhone/raop_play/Makefile.in	2008-12-06 22:45:54.000000000 +0000
@@ -15,7 +15,7 @@
 CFLAGS=-Wall
 OBJS := raop_play.o raop_client.o rtsp_client.o aexcl_lib.o base64.o aes.o m4a_stream.o \
 audio_stream.o wav_stream.o mp3_stream.o flac_stream.o ogg_stream.o aac_stream.o pls_stream.o \
-pcm_stream.o flac_stream.o
+pcm_stream.o flac_stream.o getline.o
 
 all: $(TARGET)
 
diff -ru raop_play-0.5.1/rendezvous/Makefile.in raop_play-0.5.1+iPhone/rendezvous/Makefile.in
--- raop_play-0.5.1/rendezvous/Makefile.in	2005-12-16 14:17:02.000000000 +0000
+++ raop_play-0.5.1+iPhone/rendezvous/Makefile.in	2008-12-06 22:45:48.000000000 +0000
@@ -17,7 +17,7 @@
 all: $(TARGET)
 
 
-mDNSClient: mDNSPosix.o mDNSUNP.o ExampleClientApp.o mDNS.o Client.o
+mDNSClient: mDNSPosix.o mDNSUNP.o ExampleClientApp.o mDNS.o Client.o getline.o
 	$(CC) $(LFLAGS) $^ -o $@
 
 install:
diff -ru raop_play-0.5.1/raop_play/pls_stream.c raop_play-0.5.1+iPhone/raop_play/pls_stream.c
--- raop_play-0.5.1/raop_play/pls_stream.c	2005-12-16 14:17:01.000000000 +0000
+++ raop_play-0.5.1+iPhone/raop_play/pls_stream.c	2008-12-06 22:46:54.000000000 +0000
@@ -27,6 +27,7 @@
 #include "audio_stream.h"
 #include "pls_stream.h"
 #include "aexcl_lib.h"
+#include "getline.h"
 
 
 
diff -ru raop_play-0.5.1/rendezvous/Client.c raop_play-0.5.1+iPhone/rendezvous/Client.c
--- raop_play-0.5.1/rendezvous/Client.c	2005-12-16 14:17:02.000000000 +0000
+++ raop_play-0.5.1+iPhone/rendezvous/Client.c	2008-12-06 22:46:50.000000000 +0000
@@ -89,6 +89,7 @@
 #include "mDNSPosix.h"    // Defines the specific types needed to run mDNS on this platform
 #include "ExampleClientApp.h"
 #include "mDNS.h"
+#include "getline.h"
 
 
 // Globals
