diff -ru raop_play-0.5.1/config.h.in raop_play-0.5.1+iPhone/config.h.in
--- raop_play-0.5.1/config.h.in	2005-12-16 14:17:03.000000000 +0000
+++ raop_play-0.5.1+iPhone/config.h.in	2008-12-06 03:45:42.000000000 +0000
@@ -42,7 +42,7 @@
 /* Define to 1 if you have the `samplerate' library (-lsamplerate). */
 #undef HAVE_LIBSAMPLERATE
 
-/* Define to 1 if you have the `ssl' library (-lssl). */
+/* Define to 1 if you have the `ssl' library (-lcrypto). */
 #undef HAVE_LIBSSL
 
 /* Define to 1 if your system has a GNU libc compatible `malloc' function, and
diff -ru raop_play-0.5.1/configure raop_play-0.5.1+iPhone/configure
--- raop_play-0.5.1/configure	2005-12-16 14:17:03.000000000 +0000
+++ raop_play-0.5.1+iPhone/configure	2008-12-06 03:45:45.000000000 +0000
@@ -2579,13 +2579,13 @@
 # Checks for libraries.
 
 
-echo "$as_me:$LINENO: checking for RSA_new in -lssl" >&5
-echo $ECHO_N "checking for RSA_new in -lssl... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for RSA_new in -lcrypto" >&5
+echo $ECHO_N "checking for RSA_new in -lcrypto... $ECHO_C" >&6
 if test "${ac_cv_lib_ssl_RSA_new+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lssl  $LIBS"
+LIBS="-lcrypto  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -2647,7 +2647,7 @@
 #define HAVE_LIBSSL 1
 _ACEOF
 
-  LIBS="-lssl $LIBS"
+  LIBS="-lcrypto $LIBS"
 
 else
 
diff -ru raop_play-0.5.1/raop_play/Makefile.in raop_play-0.5.1+iPhone/raop_play/Makefile.in
--- raop_play-0.5.1/raop_play/Makefile.in	2005-12-16 14:17:00.000000000 +0000
+++ raop_play-0.5.1+iPhone/raop_play/Makefile.in	2008-12-06 03:45:42.000000000 +0000
@@ -20,7 +20,7 @@
 all: $(TARGET)
 
 raop_play: $(OBJS)
-	$(CC) -o $@  -lssl -lsamplerate -lid3tag $^
+	$(CC) -o $@  -lcrypto -lsamplerate -lid3tag $^
 
 install:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)/
