diff -ru javasqlite-20080130/SQLite/JDBC2z/JDBCResultSet.java javasqlite-20080130+iPhone/SQLite/JDBC2z/JDBCResultSet.java
--- javasqlite-20080130/SQLite/JDBC2z/JDBCResultSet.java	2008-01-08 10:03:08.000000000 +0000
+++ javasqlite-20080130+iPhone/SQLite/JDBC2z/JDBCResultSet.java	2008-03-11 02:55:52.000000000 +0000
@@ -74,7 +74,7 @@
     public JDBCResultSet(SQLite.TableResult tr, JDBCStatement s) {
 	this.tr = tr;
 	this.s = s;
-	this.m = null;
+	this.md = null;
 	this.lastg = null;
 	this.row = -1;
 	this.updatable = UPD_UNKNOWN;
