--- apt/methods/http.cc	2018-07-30 21:45:07.000000000 -1000
+++ apt+iPhone/methods/http.cc	2018-07-30 23:55:04.000000000 -1000
@@ -603,7 +603,7 @@
 
    if (UniqueID_ == NULL)
    if (void *lockdown = lockdown_connect()) {
-      UniqueID_ = lockdown_copy_value(lockdown, NULL, kLockdownUniqueDeviceIDKey);
+      UniqueID_ = (CFStringRef)lockdown_copy_value(lockdown, NULL, kLockdownUniqueDeviceIDKey);
       lockdown_disconnect(lockdown);
    }
 
