--- apt/apt-pkg/deb/debmetaindex.cc	2018-07-30 21:45:07.000000000 -1000
+++ apt+iPhone/apt-pkg/deb/debmetaindex.cc	2018-08-02 00:03:17.000000000 -1000
@@ -458,9 +458,6 @@
 
    if (CheckValidUntil == true)
    {
-      if (Date == 0)
-          _error->Warning( _("Invalid '%s' entry in Release file %s"), "Date", Filename.c_str());
-
       std::string const Label = Section.FindS("Label");
       std::string const StrValidUntil = Section.FindS("Valid-Until");
 
