--- apt/apt-pkg/acquire.cc	2018-07-30 21:45:07.000000000 -1000
+++ apt+iPhone/apt-pkg/acquire.cc	2018-08-02 00:04:25.000000000 -1000
@@ -603,7 +603,6 @@
    struct passwd const * const pw = getpwnam(SandboxUser.c_str());
    if (pw == NULL)
    {
-      _error->Warning(_("No sandbox user '%s' on the system, can not drop privileges"), SandboxUser.c_str());
       _config->Set("APT::Sandbox::User", "");
       return;
    }
