diff -ru php-5.2.8/main/php.h php-5.2.8+iPhone/main/php.h
--- php-5.2.8/main/php.h	2007-12-31 07:20:15.000000000 +0000
+++ php-5.2.8+iPhone/main/php.h	2009-01-18 09:43:06.000000000 +0000
@@ -418,7 +418,7 @@
  */
 
 #ifndef XtOffset
-#if defined(CRAY) || (defined(__arm) && !(defined(LINUX) || defined(__riscos__)))
+#if defined(CRAY) || (defined(__arm) && !(defined(LINUX) || defined(__riscos__) || defined(__APPLE__)))
 #ifdef __STDC__
 #define XtOffset(p_type, field) _Offsetof(p_type, field)
 #else
diff -ru php-5.2.8/Zend/zend_ini.h php-5.2.8+iPhone/Zend/zend_ini.h
--- php-5.2.8/Zend/zend_ini.h	2007-12-31 07:20:03.000000000 +0000
+++ php-5.2.8+iPhone/Zend/zend_ini.h	2009-01-18 09:43:16.000000000 +0000
@@ -28,7 +28,7 @@
 #define ZEND_INI_ALL (ZEND_INI_USER|ZEND_INI_PERDIR|ZEND_INI_SYSTEM)
 
 #ifndef XtOffsetOf
-# if defined(CRAY) || (defined(__arm) && !defined(LINUX))
+# if defined(CRAY) || (defined(__arm) && !(defined(LINUX) || defined(__riscos__) || defined(__APPLE__)))
 # ifdef __STDC__
 # define XtOffset(p_type, field) _Offsetof(p_type, field)
 # else
