diff -r -u dictd-1.9.9/libmaa/sl.c dictd-1.9.9+iPhone/libmaa/sl.c
--- dictd-1.9.9/libmaa/sl.c	2002-08-02 19:43:15.000000000 +0000
+++ dictd-1.9.9+iPhone/libmaa/sl.c	2008-04-05 17:27:43.000000000 +0000
@@ -355,6 +355,7 @@
    _sl_check( list );
 }
 
+#if 0
 /* \doc Find the datum in |list| that has an associated value of |key|.
    Return that datum (a pointer), or "NULL" if the |key| is not found. */
 
@@ -371,6 +372,7 @@
    if (pt && !l->compare( l->key( pt->datum ), key )) return pt->datum;
    return NULL;
 }
+#endif
 
 /* \doc Iterate |f| over every datum in |list|.  If |f| returns non-zero,
    then abort the remainder of the iteration.  Iterations are designed to
